// @cdktf/provider-github/lib/repository-pull-request/index.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.RepositoryPullRequest = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const cdktf = require("cdktf");
/**
* Represents a {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/repository_pull_request github_repository_pull_request}
*/
class RepositoryPullRequest extends cdktf.TerraformResource {
    // ==============
    // STATIC Methods
    // ==============
    /**
    * Generates CDKTF code for importing a RepositoryPullRequest resource upon running "cdktf plan <stack-name>"
    * @param scope The scope in which to define this construct
    * @param importToId The construct id used in the generated config for the RepositoryPullRequest to import
    * @param importFromId The id of the existing RepositoryPullRequest that should be imported. Refer to the {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/repository_pull_request#import import section} in the documentation of this resource for the id to use
    * @param provider? Optional instance of the provider where the RepositoryPullRequest to import is found
    */
    static generateConfigForImport(scope, importToId, importFromId, provider) {
        return new cdktf.ImportableResource(scope, importToId, { terraformResourceType: "github_repository_pull_request", importId: importFromId, provider });
    }
    // ===========
    // INITIALIZER
    // ===========
    /**
    * Create a new {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/repository_pull_request github_repository_pull_request} Resource
    *
    * @param scope The scope in which to define this construct
    * @param id The scoped construct ID. Must be unique amongst siblings in the same scope
    * @param options RepositoryPullRequestConfig
    */
    constructor(scope, id, config) {
        super(scope, id, {
            terraformResourceType: 'github_repository_pull_request',
            terraformGeneratorMetadata: {
                providerName: 'github',
                providerVersion: '5.40.0',
                providerVersionConstraint: '~> 5.0'
            },
            provider: config.provider,
            dependsOn: config.dependsOn,
            count: config.count,
            lifecycle: config.lifecycle,
            provisioners: config.provisioners,
            connection: config.connection,
            forEach: config.forEach
        });
        this._baseRef = config.baseRef;
        this._baseRepository = config.baseRepository;
        this._body = config.body;
        this._headRef = config.headRef;
        this._id = config.id;
        this._maintainerCanModify = config.maintainerCanModify;
        this._owner = config.owner;
        this._title = config.title;
    }
    get baseRef() {
        return this.getStringAttribute('base_ref');
    }
    set baseRef(value) {
        this._baseRef = value;
    }
    // Temporarily expose input value. Use with caution.
    get baseRefInput() {
        return this._baseRef;
    }
    get baseRepository() {
        return this.getStringAttribute('base_repository');
    }
    set baseRepository(value) {
        this._baseRepository = value;
    }
    // Temporarily expose input value. Use with caution.
    get baseRepositoryInput() {
        return this._baseRepository;
    }
    // base_sha - computed: true, optional: false, required: false
    get baseSha() {
        return this.getStringAttribute('base_sha');
    }
    get body() {
        return this.getStringAttribute('body');
    }
    set body(value) {
        this._body = value;
    }
    resetBody() {
        this._body = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get bodyInput() {
        return this._body;
    }
    // draft - computed: true, optional: false, required: false
    get draft() {
        return this.getBooleanAttribute('draft');
    }
    get headRef() {
        return this.getStringAttribute('head_ref');
    }
    set headRef(value) {
        this._headRef = value;
    }
    // Temporarily expose input value. Use with caution.
    get headRefInput() {
        return this._headRef;
    }
    // head_sha - computed: true, optional: false, required: false
    get headSha() {
        return this.getStringAttribute('head_sha');
    }
    get id() {
        return this.getStringAttribute('id');
    }
    set id(value) {
        this._id = value;
    }
    resetId() {
        this._id = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get idInput() {
        return this._id;
    }
    // labels - computed: true, optional: false, required: false
    get labels() {
        return this.getListAttribute('labels');
    }
    get maintainerCanModify() {
        return this.getBooleanAttribute('maintainer_can_modify');
    }
    set maintainerCanModify(value) {
        this._maintainerCanModify = value;
    }
    resetMaintainerCanModify() {
        this._maintainerCanModify = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get maintainerCanModifyInput() {
        return this._maintainerCanModify;
    }
    // number - computed: true, optional: false, required: false
    get number() {
        return this.getNumberAttribute('number');
    }
    // opened_at - computed: true, optional: false, required: false
    get openedAt() {
        return this.getNumberAttribute('opened_at');
    }
    // opened_by - computed: true, optional: false, required: false
    get openedBy() {
        return this.getStringAttribute('opened_by');
    }
    get owner() {
        return this.getStringAttribute('owner');
    }
    set owner(value) {
        this._owner = value;
    }
    resetOwner() {
        this._owner = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get ownerInput() {
        return this._owner;
    }
    // state - computed: true, optional: false, required: false
    get state() {
        return this.getStringAttribute('state');
    }
    get title() {
        return this.getStringAttribute('title');
    }
    set title(value) {
        this._title = value;
    }
    // Temporarily expose input value. Use with caution.
    get titleInput() {
        return this._title;
    }
    // updated_at - computed: true, optional: false, required: false
    get updatedAt() {
        return this.getNumberAttribute('updated_at');
    }
    // =========
    // SYNTHESIS
    // =========
    synthesizeAttributes() {
        return {
            base_ref: cdktf.stringToTerraform(this._baseRef),
            base_repository: cdktf.stringToTerraform(this._baseRepository),
            body: cdktf.stringToTerraform(this._body),
            head_ref: cdktf.stringToTerraform(this._headRef),
            id: cdktf.stringToTerraform(this._id),
            maintainer_can_modify: cdktf.booleanToTerraform(this._maintainerCanModify),
            owner: cdktf.stringToTerraform(this._owner),
            title: cdktf.stringToTerraform(this._title),
        };
    }
}
exports.RepositoryPullRequest = RepositoryPullRequest;
_a = JSII_RTTI_SYMBOL_1;
RepositoryPullRequest[_a] = { fqn: "@cdktf/provider-github.repositoryPullRequest.RepositoryPullRequest", version: "12.0.0" };
// =================
// STATIC PROPERTIES
// =================
RepositoryPullRequest.tfResourceType = "github_repository_pull_request";
