// @cdktf/provider-github/lib/data-github-users/index.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataGithubUsers = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const cdktf = require("cdktf");
/**
* Represents a {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/users github_users}
*/
class DataGithubUsers extends cdktf.TerraformDataSource {
    // ==============
    // STATIC Methods
    // ==============
    /**
    * Generates CDKTF code for importing a DataGithubUsers resource upon running "cdktf plan <stack-name>"
    * @param scope The scope in which to define this construct
    * @param importToId The construct id used in the generated config for the DataGithubUsers to import
    * @param importFromId The id of the existing DataGithubUsers that should be imported. Refer to the {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/users#import import section} in the documentation of this resource for the id to use
    * @param provider? Optional instance of the provider where the DataGithubUsers to import is found
    */
    static generateConfigForImport(scope, importToId, importFromId, provider) {
        return new cdktf.ImportableResource(scope, importToId, { terraformResourceType: "github_users", importId: importFromId, provider });
    }
    // ===========
    // INITIALIZER
    // ===========
    /**
    * Create a new {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/users github_users} Data Source
    *
    * @param scope The scope in which to define this construct
    * @param id The scoped construct ID. Must be unique amongst siblings in the same scope
    * @param options DataGithubUsersConfig
    */
    constructor(scope, id, config) {
        super(scope, id, {
            terraformResourceType: 'github_users',
            terraformGeneratorMetadata: {
                providerName: 'github',
                providerVersion: '5.40.0',
                providerVersionConstraint: '~> 5.0'
            },
            provider: config.provider,
            dependsOn: config.dependsOn,
            count: config.count,
            lifecycle: config.lifecycle,
            provisioners: config.provisioners,
            connection: config.connection,
            forEach: config.forEach
        });
        this._id = config.id;
        this._usernames = config.usernames;
    }
    // ==========
    // ATTRIBUTES
    // ==========
    // emails - computed: true, optional: false, required: false
    get emails() {
        return this.getListAttribute('emails');
    }
    get id() {
        return this.getStringAttribute('id');
    }
    set id(value) {
        this._id = value;
    }
    resetId() {
        this._id = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get idInput() {
        return this._id;
    }
    // logins - computed: true, optional: false, required: false
    get logins() {
        return this.getListAttribute('logins');
    }
    // node_ids - computed: true, optional: false, required: false
    get nodeIds() {
        return this.getListAttribute('node_ids');
    }
    // unknown_logins - computed: true, optional: false, required: false
    get unknownLogins() {
        return this.getListAttribute('unknown_logins');
    }
    get usernames() {
        return this.getListAttribute('usernames');
    }
    set usernames(value) {
        this._usernames = value;
    }
    // Temporarily expose input value. Use with caution.
    get usernamesInput() {
        return this._usernames;
    }
    // =========
    // SYNTHESIS
    // =========
    synthesizeAttributes() {
        return {
            id: cdktf.stringToTerraform(this._id),
            usernames: cdktf.listMapper(cdktf.stringToTerraform, false)(this._usernames),
        };
    }
}
exports.DataGithubUsers = DataGithubUsers;
_a = JSII_RTTI_SYMBOL_1;
DataGithubUsers[_a] = { fqn: "@cdktf/provider-github.dataGithubUsers.DataGithubUsers", version: "12.0.0" };
// =================
// STATIC PROPERTIES
// =================
DataGithubUsers.tfResourceType = "github_users";
