// @cdktf/provider-github/lib/actions-repository-permissions/index.js
"use strict";
var _a, _b;
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionsRepositoryPermissions = exports.ActionsRepositoryPermissionsAllowedActionsConfigOutputReference = exports.actionsRepositoryPermissionsAllowedActionsConfigToTerraform = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const cdktf = require("cdktf");
function actionsRepositoryPermissionsAllowedActionsConfigToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        github_owned_allowed: cdktf.booleanToTerraform(struct.githubOwnedAllowed),
        patterns_allowed: cdktf.listMapper(cdktf.stringToTerraform, false)(struct.patternsAllowed),
        verified_allowed: cdktf.booleanToTerraform(struct.verifiedAllowed),
    };
}
exports.actionsRepositoryPermissionsAllowedActionsConfigToTerraform = actionsRepositoryPermissionsAllowedActionsConfigToTerraform;
class ActionsRepositoryPermissionsAllowedActionsConfigOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._githubOwnedAllowed !== undefined) {
            hasAnyValues = true;
            internalValueResult.githubOwnedAllowed = this._githubOwnedAllowed;
        }
        if (this._patternsAllowed !== undefined) {
            hasAnyValues = true;
            internalValueResult.patternsAllowed = this._patternsAllowed;
        }
        if (this._verifiedAllowed !== undefined) {
            hasAnyValues = true;
            internalValueResult.verifiedAllowed = this._verifiedAllowed;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._githubOwnedAllowed = undefined;
            this._patternsAllowed = undefined;
            this._verifiedAllowed = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._githubOwnedAllowed = value.githubOwnedAllowed;
            this._patternsAllowed = value.patternsAllowed;
            this._verifiedAllowed = value.verifiedAllowed;
        }
    }
    get githubOwnedAllowed() {
        return this.getBooleanAttribute('github_owned_allowed');
    }
    set githubOwnedAllowed(value) {
        this._githubOwnedAllowed = value;
    }
    // Temporarily expose input value. Use with caution.
    get githubOwnedAllowedInput() {
        return this._githubOwnedAllowed;
    }
    get patternsAllowed() {
        return cdktf.Fn.tolist(this.getListAttribute('patterns_allowed'));
    }
    set patternsAllowed(value) {
        this._patternsAllowed = value;
    }
    resetPatternsAllowed() {
        this._patternsAllowed = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get patternsAllowedInput() {
        return this._patternsAllowed;
    }
    get verifiedAllowed() {
        return this.getBooleanAttribute('verified_allowed');
    }
    set verifiedAllowed(value) {
        this._verifiedAllowed = value;
    }
    resetVerifiedAllowed() {
        this._verifiedAllowed = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get verifiedAllowedInput() {
        return this._verifiedAllowed;
    }
}
exports.ActionsRepositoryPermissionsAllowedActionsConfigOutputReference = ActionsRepositoryPermissionsAllowedActionsConfigOutputReference;
_a = JSII_RTTI_SYMBOL_1;
ActionsRepositoryPermissionsAllowedActionsConfigOutputReference[_a] = { fqn: "@cdktf/provider-github.actionsRepositoryPermissions.ActionsRepositoryPermissionsAllowedActionsConfigOutputReference", version: "12.0.0" };
/**
* Represents a {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/actions_repository_permissions github_actions_repository_permissions}
*/
class ActionsRepositoryPermissions extends cdktf.TerraformResource {
    // ==============
    // STATIC Methods
    // ==============
    /**
    * Generates CDKTF code for importing a ActionsRepositoryPermissions resource upon running "cdktf plan <stack-name>"
    * @param scope The scope in which to define this construct
    * @param importToId The construct id used in the generated config for the ActionsRepositoryPermissions to import
    * @param importFromId The id of the existing ActionsRepositoryPermissions that should be imported. Refer to the {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/actions_repository_permissions#import import section} in the documentation of this resource for the id to use
    * @param provider? Optional instance of the provider where the ActionsRepositoryPermissions to import is found
    */
    static generateConfigForImport(scope, importToId, importFromId, provider) {
        return new cdktf.ImportableResource(scope, importToId, { terraformResourceType: "github_actions_repository_permissions", importId: importFromId, provider });
    }
    // ===========
    // INITIALIZER
    // ===========
    /**
    * Create a new {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/actions_repository_permissions github_actions_repository_permissions} Resource
    *
    * @param scope The scope in which to define this construct
    * @param id The scoped construct ID. Must be unique amongst siblings in the same scope
    * @param options ActionsRepositoryPermissionsConfig
    */
    constructor(scope, id, config) {
        super(scope, id, {
            terraformResourceType: 'github_actions_repository_permissions',
            terraformGeneratorMetadata: {
                providerName: 'github',
                providerVersion: '5.40.0',
                providerVersionConstraint: '~> 5.0'
            },
            provider: config.provider,
            dependsOn: config.dependsOn,
            count: config.count,
            lifecycle: config.lifecycle,
            provisioners: config.provisioners,
            connection: config.connection,
            forEach: config.forEach
        });
        // allowed_actions_config - computed: false, optional: true, required: false
        this._allowedActionsConfig = new ActionsRepositoryPermissionsAllowedActionsConfigOutputReference(this, "allowed_actions_config");
        this._allowedActions = config.allowedActions;
        this._enabled = config.enabled;
        this._id = config.id;
        this._repository = config.repository;
        this._allowedActionsConfig.internalValue = config.allowedActionsConfig;
    }
    get allowedActions() {
        return this.getStringAttribute('allowed_actions');
    }
    set allowedActions(value) {
        this._allowedActions = value;
    }
    resetAllowedActions() {
        this._allowedActions = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get allowedActionsInput() {
        return this._allowedActions;
    }
    get enabled() {
        return this.getBooleanAttribute('enabled');
    }
    set enabled(value) {
        this._enabled = value;
    }
    resetEnabled() {
        this._enabled = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get enabledInput() {
        return this._enabled;
    }
    get id() {
        return this.getStringAttribute('id');
    }
    set id(value) {
        this._id = value;
    }
    resetId() {
        this._id = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get idInput() {
        return this._id;
    }
    get repository() {
        return this.getStringAttribute('repository');
    }
    set repository(value) {
        this._repository = value;
    }
    // Temporarily expose input value. Use with caution.
    get repositoryInput() {
        return this._repository;
    }
    get allowedActionsConfig() {
        return this._allowedActionsConfig;
    }
    putAllowedActionsConfig(value) {
        this._allowedActionsConfig.internalValue = value;
    }
    resetAllowedActionsConfig() {
        this._allowedActionsConfig.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get allowedActionsConfigInput() {
        return this._allowedActionsConfig.internalValue;
    }
    // =========
    // SYNTHESIS
    // =========
    synthesizeAttributes() {
        return {
            allowed_actions: cdktf.stringToTerraform(this._allowedActions),
            enabled: cdktf.booleanToTerraform(this._enabled),
            id: cdktf.stringToTerraform(this._id),
            repository: cdktf.stringToTerraform(this._repository),
            allowed_actions_config: actionsRepositoryPermissionsAllowedActionsConfigToTerraform(this._allowedActionsConfig.internalValue),
        };
    }
}
exports.ActionsRepositoryPermissions = ActionsRepositoryPermissions;
_b = JSII_RTTI_SYMBOL_1;
ActionsRepositoryPermissions[_b] = { fqn: "@cdktf/provider-github.actionsRepositoryPermissions.ActionsRepositoryPermissions", version: "12.0.0" };
// =================
// STATIC PROPERTIES
// =================
ActionsRepositoryPermissions.tfResourceType = "github_actions_repository_permissions";
