// @cdktf/provider-github/lib/data-github-repository-deployment-branch-policies/index.js
"use strict";
var _a, _b, _c;
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataGithubRepositoryDeploymentBranchPolicies = exports.DataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesList = exports.DataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesOutputReference = exports.dataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesToTerraform = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const cdktf = require("cdktf");
function dataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {};
}
exports.dataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesToTerraform = dataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesToTerraform;
class DataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    * @param complexObjectIndex the index of this item in the list
    * @param complexObjectIsFromSet whether the list is wrapping a set (will add tolist() to be able to access an item via an index)
    */
    constructor(terraformResource, terraformAttribute, complexObjectIndex, complexObjectIsFromSet) {
        super(terraformResource, terraformAttribute, complexObjectIsFromSet, complexObjectIndex);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
        }
    }
    // id - computed: true, optional: false, required: false
    get id() {
        return this.getStringAttribute('id');
    }
    // name - computed: true, optional: false, required: false
    get name() {
        return this.getStringAttribute('name');
    }
}
exports.DataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesOutputReference = DataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesOutputReference;
_a = JSII_RTTI_SYMBOL_1;
DataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesOutputReference[_a] = { fqn: "@cdktf/provider-github.dataGithubRepositoryDeploymentBranchPolicies.DataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesOutputReference", version: "12.0.0" };
class DataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesList extends cdktf.ComplexList {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    * @param wrapsSet whether the list is wrapping a set (will add tolist() to be able to access an item via an index)
    */
    constructor(terraformResource, terraformAttribute, wrapsSet) {
        super(terraformResource, terraformAttribute, wrapsSet);
        this.terraformResource = terraformResource;
        this.terraformAttribute = terraformAttribute;
        this.wrapsSet = wrapsSet;
    }
    /**
    * @param index the index of the item to return
    */
    get(index) {
        return new DataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesOutputReference(this.terraformResource, this.terraformAttribute, index, this.wrapsSet);
    }
}
exports.DataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesList = DataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesList;
_b = JSII_RTTI_SYMBOL_1;
DataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesList[_b] = { fqn: "@cdktf/provider-github.dataGithubRepositoryDeploymentBranchPolicies.DataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesList", version: "12.0.0" };
/**
* Represents a {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/repository_deployment_branch_policies github_repository_deployment_branch_policies}
*/
class DataGithubRepositoryDeploymentBranchPolicies extends cdktf.TerraformDataSource {
    // ==============
    // STATIC Methods
    // ==============
    /**
    * Generates CDKTF code for importing a DataGithubRepositoryDeploymentBranchPolicies resource upon running "cdktf plan <stack-name>"
    * @param scope The scope in which to define this construct
    * @param importToId The construct id used in the generated config for the DataGithubRepositoryDeploymentBranchPolicies to import
    * @param importFromId The id of the existing DataGithubRepositoryDeploymentBranchPolicies that should be imported. Refer to the {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/repository_deployment_branch_policies#import import section} in the documentation of this resource for the id to use
    * @param provider? Optional instance of the provider where the DataGithubRepositoryDeploymentBranchPolicies to import is found
    */
    static generateConfigForImport(scope, importToId, importFromId, provider) {
        return new cdktf.ImportableResource(scope, importToId, { terraformResourceType: "github_repository_deployment_branch_policies", importId: importFromId, provider });
    }
    // ===========
    // INITIALIZER
    // ===========
    /**
    * Create a new {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/repository_deployment_branch_policies github_repository_deployment_branch_policies} Data Source
    *
    * @param scope The scope in which to define this construct
    * @param id The scoped construct ID. Must be unique amongst siblings in the same scope
    * @param options DataGithubRepositoryDeploymentBranchPoliciesConfig
    */
    constructor(scope, id, config) {
        super(scope, id, {
            terraformResourceType: 'github_repository_deployment_branch_policies',
            terraformGeneratorMetadata: {
                providerName: 'github',
                providerVersion: '5.40.0',
                providerVersionConstraint: '~> 5.0'
            },
            provider: config.provider,
            dependsOn: config.dependsOn,
            count: config.count,
            lifecycle: config.lifecycle,
            provisioners: config.provisioners,
            connection: config.connection,
            forEach: config.forEach
        });
        // ==========
        // ATTRIBUTES
        // ==========
        // deployment_branch_policies - computed: true, optional: false, required: false
        this._deploymentBranchPolicies = new DataGithubRepositoryDeploymentBranchPoliciesDeploymentBranchPoliciesList(this, "deployment_branch_policies", false);
        this._environmentName = config.environmentName;
        this._id = config.id;
        this._repository = config.repository;
    }
    get deploymentBranchPolicies() {
        return this._deploymentBranchPolicies;
    }
    get environmentName() {
        return this.getStringAttribute('environment_name');
    }
    set environmentName(value) {
        this._environmentName = value;
    }
    // Temporarily expose input value. Use with caution.
    get environmentNameInput() {
        return this._environmentName;
    }
    get id() {
        return this.getStringAttribute('id');
    }
    set id(value) {
        this._id = value;
    }
    resetId() {
        this._id = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get idInput() {
        return this._id;
    }
    get repository() {
        return this.getStringAttribute('repository');
    }
    set repository(value) {
        this._repository = value;
    }
    // Temporarily expose input value. Use with caution.
    get repositoryInput() {
        return this._repository;
    }
    // =========
    // SYNTHESIS
    // =========
    synthesizeAttributes() {
        return {
            environment_name: cdktf.stringToTerraform(this._environmentName),
            id: cdktf.stringToTerraform(this._id),
            repository: cdktf.stringToTerraform(this._repository),
        };
    }
}
exports.DataGithubRepositoryDeploymentBranchPolicies = DataGithubRepositoryDeploymentBranchPolicies;
_c = JSII_RTTI_SYMBOL_1;
DataGithubRepositoryDeploymentBranchPolicies[_c] = { fqn: "@cdktf/provider-github.dataGithubRepositoryDeploymentBranchPolicies.DataGithubRepositoryDeploymentBranchPolicies", version: "12.0.0" };
// =================
// STATIC PROPERTIES
// =================
DataGithubRepositoryDeploymentBranchPolicies.tfResourceType = "github_repository_deployment_branch_policies";
