// @cdktf/provider-github/lib/user-invitation-accepter/index.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserInvitationAccepter = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const cdktf = require("cdktf");
/**
* Represents a {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/user_invitation_accepter github_user_invitation_accepter}
*/
class UserInvitationAccepter extends cdktf.TerraformResource {
    // ==============
    // STATIC Methods
    // ==============
    /**
    * Generates CDKTF code for importing a UserInvitationAccepter resource upon running "cdktf plan <stack-name>"
    * @param scope The scope in which to define this construct
    * @param importToId The construct id used in the generated config for the UserInvitationAccepter to import
    * @param importFromId The id of the existing UserInvitationAccepter that should be imported. Refer to the {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/user_invitation_accepter#import import section} in the documentation of this resource for the id to use
    * @param provider? Optional instance of the provider where the UserInvitationAccepter to import is found
    */
    static generateConfigForImport(scope, importToId, importFromId, provider) {
        return new cdktf.ImportableResource(scope, importToId, { terraformResourceType: "github_user_invitation_accepter", importId: importFromId, provider });
    }
    // ===========
    // INITIALIZER
    // ===========
    /**
    * Create a new {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/user_invitation_accepter github_user_invitation_accepter} Resource
    *
    * @param scope The scope in which to define this construct
    * @param id The scoped construct ID. Must be unique amongst siblings in the same scope
    * @param options UserInvitationAccepterConfig = {}
    */
    constructor(scope, id, config = {}) {
        super(scope, id, {
            terraformResourceType: 'github_user_invitation_accepter',
            terraformGeneratorMetadata: {
                providerName: 'github',
                providerVersion: '5.40.0',
                providerVersionConstraint: '~> 5.0'
            },
            provider: config.provider,
            dependsOn: config.dependsOn,
            count: config.count,
            lifecycle: config.lifecycle,
            provisioners: config.provisioners,
            connection: config.connection,
            forEach: config.forEach
        });
        this._allowEmptyId = config.allowEmptyId;
        this._id = config.id;
        this._invitationId = config.invitationId;
    }
    get allowEmptyId() {
        return this.getBooleanAttribute('allow_empty_id');
    }
    set allowEmptyId(value) {
        this._allowEmptyId = value;
    }
    resetAllowEmptyId() {
        this._allowEmptyId = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get allowEmptyIdInput() {
        return this._allowEmptyId;
    }
    get id() {
        return this.getStringAttribute('id');
    }
    set id(value) {
        this._id = value;
    }
    resetId() {
        this._id = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get idInput() {
        return this._id;
    }
    get invitationId() {
        return this.getStringAttribute('invitation_id');
    }
    set invitationId(value) {
        this._invitationId = value;
    }
    resetInvitationId() {
        this._invitationId = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get invitationIdInput() {
        return this._invitationId;
    }
    // =========
    // SYNTHESIS
    // =========
    synthesizeAttributes() {
        return {
            allow_empty_id: cdktf.booleanToTerraform(this._allowEmptyId),
            id: cdktf.stringToTerraform(this._id),
            invitation_id: cdktf.stringToTerraform(this._invitationId),
        };
    }
}
exports.UserInvitationAccepter = UserInvitationAccepter;
_a = JSII_RTTI_SYMBOL_1;
UserInvitationAccepter[_a] = { fqn: "@cdktf/provider-github.userInvitationAccepter.UserInvitationAccepter", version: "12.0.0" };
// =================
// STATIC PROPERTIES
// =================
UserInvitationAccepter.tfResourceType = "github_user_invitation_accepter";
