// @cdktf/provider-github/lib/data-github-repositories/index.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataGithubRepositories = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const cdktf = require("cdktf");
/**
* Represents a {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/repositories github_repositories}
*/
class DataGithubRepositories extends cdktf.TerraformDataSource {
    // ==============
    // STATIC Methods
    // ==============
    /**
    * Generates CDKTF code for importing a DataGithubRepositories resource upon running "cdktf plan <stack-name>"
    * @param scope The scope in which to define this construct
    * @param importToId The construct id used in the generated config for the DataGithubRepositories to import
    * @param importFromId The id of the existing DataGithubRepositories that should be imported. Refer to the {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/repositories#import import section} in the documentation of this resource for the id to use
    * @param provider? Optional instance of the provider where the DataGithubRepositories to import is found
    */
    static generateConfigForImport(scope, importToId, importFromId, provider) {
        return new cdktf.ImportableResource(scope, importToId, { terraformResourceType: "github_repositories", importId: importFromId, provider });
    }
    // ===========
    // INITIALIZER
    // ===========
    /**
    * Create a new {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/repositories github_repositories} Data Source
    *
    * @param scope The scope in which to define this construct
    * @param id The scoped construct ID. Must be unique amongst siblings in the same scope
    * @param options DataGithubRepositoriesConfig
    */
    constructor(scope, id, config) {
        super(scope, id, {
            terraformResourceType: 'github_repositories',
            terraformGeneratorMetadata: {
                providerName: 'github',
                providerVersion: '5.40.0',
                providerVersionConstraint: '~> 5.0'
            },
            provider: config.provider,
            dependsOn: config.dependsOn,
            count: config.count,
            lifecycle: config.lifecycle,
            provisioners: config.provisioners,
            connection: config.connection,
            forEach: config.forEach
        });
        this._id = config.id;
        this._includeRepoId = config.includeRepoId;
        this._query = config.query;
        this._resultsPerPage = config.resultsPerPage;
        this._sort = config.sort;
    }
    // ==========
    // ATTRIBUTES
    // ==========
    // full_names - computed: true, optional: false, required: false
    get fullNames() {
        return this.getListAttribute('full_names');
    }
    get id() {
        return this.getStringAttribute('id');
    }
    set id(value) {
        this._id = value;
    }
    resetId() {
        this._id = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get idInput() {
        return this._id;
    }
    get includeRepoId() {
        return this.getBooleanAttribute('include_repo_id');
    }
    set includeRepoId(value) {
        this._includeRepoId = value;
    }
    resetIncludeRepoId() {
        this._includeRepoId = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get includeRepoIdInput() {
        return this._includeRepoId;
    }
    // names - computed: true, optional: false, required: false
    get names() {
        return this.getListAttribute('names');
    }
    get query() {
        return this.getStringAttribute('query');
    }
    set query(value) {
        this._query = value;
    }
    // Temporarily expose input value. Use with caution.
    get queryInput() {
        return this._query;
    }
    // repo_ids - computed: true, optional: false, required: false
    get repoIds() {
        return this.getNumberListAttribute('repo_ids');
    }
    get resultsPerPage() {
        return this.getNumberAttribute('results_per_page');
    }
    set resultsPerPage(value) {
        this._resultsPerPage = value;
    }
    resetResultsPerPage() {
        this._resultsPerPage = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get resultsPerPageInput() {
        return this._resultsPerPage;
    }
    get sort() {
        return this.getStringAttribute('sort');
    }
    set sort(value) {
        this._sort = value;
    }
    resetSort() {
        this._sort = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get sortInput() {
        return this._sort;
    }
    // =========
    // SYNTHESIS
    // =========
    synthesizeAttributes() {
        return {
            id: cdktf.stringToTerraform(this._id),
            include_repo_id: cdktf.booleanToTerraform(this._includeRepoId),
            query: cdktf.stringToTerraform(this._query),
            results_per_page: cdktf.numberToTerraform(this._resultsPerPage),
            sort: cdktf.stringToTerraform(this._sort),
        };
    }
}
exports.DataGithubRepositories = DataGithubRepositories;
_a = JSII_RTTI_SYMBOL_1;
DataGithubRepositories[_a] = { fqn: "@cdktf/provider-github.dataGithubRepositories.DataGithubRepositories", version: "12.0.0" };
// =================
// STATIC PROPERTIES
// =================
DataGithubRepositories.tfResourceType = "github_repositories";
