// @cdktf/provider-github/lib/data-github-user-external-identity/index.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataGithubUserExternalIdentity = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const cdktf = require("cdktf");
/**
* Represents a {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/user_external_identity github_user_external_identity}
*/
class DataGithubUserExternalIdentity extends cdktf.TerraformDataSource {
    // ==============
    // STATIC Methods
    // ==============
    /**
    * Generates CDKTF code for importing a DataGithubUserExternalIdentity resource upon running "cdktf plan <stack-name>"
    * @param scope The scope in which to define this construct
    * @param importToId The construct id used in the generated config for the DataGithubUserExternalIdentity to import
    * @param importFromId The id of the existing DataGithubUserExternalIdentity that should be imported. Refer to the {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/user_external_identity#import import section} in the documentation of this resource for the id to use
    * @param provider? Optional instance of the provider where the DataGithubUserExternalIdentity to import is found
    */
    static generateConfigForImport(scope, importToId, importFromId, provider) {
        return new cdktf.ImportableResource(scope, importToId, { terraformResourceType: "github_user_external_identity", importId: importFromId, provider });
    }
    // ===========
    // INITIALIZER
    // ===========
    /**
    * Create a new {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/user_external_identity github_user_external_identity} Data Source
    *
    * @param scope The scope in which to define this construct
    * @param id The scoped construct ID. Must be unique amongst siblings in the same scope
    * @param options DataGithubUserExternalIdentityConfig
    */
    constructor(scope, id, config) {
        super(scope, id, {
            terraformResourceType: 'github_user_external_identity',
            terraformGeneratorMetadata: {
                providerName: 'github',
                providerVersion: '5.40.0',
                providerVersionConstraint: '~> 5.0'
            },
            provider: config.provider,
            dependsOn: config.dependsOn,
            count: config.count,
            lifecycle: config.lifecycle,
            provisioners: config.provisioners,
            connection: config.connection,
            forEach: config.forEach
        });
        // saml_identity - computed: true, optional: false, required: false
        this._samlIdentity = new cdktf.StringMap(this, "saml_identity");
        // scim_identity - computed: true, optional: false, required: false
        this._scimIdentity = new cdktf.StringMap(this, "scim_identity");
        this._id = config.id;
        this._username = config.username;
    }
    get id() {
        return this.getStringAttribute('id');
    }
    set id(value) {
        this._id = value;
    }
    resetId() {
        this._id = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get idInput() {
        return this._id;
    }
    // login - computed: true, optional: false, required: false
    get login() {
        return this.getStringAttribute('login');
    }
    get samlIdentity() {
        return this._samlIdentity;
    }
    get scimIdentity() {
        return this._scimIdentity;
    }
    get username() {
        return this.getStringAttribute('username');
    }
    set username(value) {
        this._username = value;
    }
    // Temporarily expose input value. Use with caution.
    get usernameInput() {
        return this._username;
    }
    // =========
    // SYNTHESIS
    // =========
    synthesizeAttributes() {
        return {
            id: cdktf.stringToTerraform(this._id),
            username: cdktf.stringToTerraform(this._username),
        };
    }
}
exports.DataGithubUserExternalIdentity = DataGithubUserExternalIdentity;
_a = JSII_RTTI_SYMBOL_1;
DataGithubUserExternalIdentity[_a] = { fqn: "@cdktf/provider-github.dataGithubUserExternalIdentity.DataGithubUserExternalIdentity", version: "12.0.0" };
// =================
// STATIC PROPERTIES
// =================
DataGithubUserExternalIdentity.tfResourceType = "github_user_external_identity";
