// @cdktf/provider-github/lib/repository-ruleset/index.js
"use strict";
var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, _o, _p, _q, _r;
Object.defineProperty(exports, "__esModule", { value: true });
exports.RepositoryRuleset = exports.RepositoryRulesetRulesOutputReference = exports.repositoryRulesetRulesToTerraform = exports.RepositoryRulesetRulesTagNamePatternOutputReference = exports.repositoryRulesetRulesTagNamePatternToTerraform = exports.RepositoryRulesetRulesRequiredStatusChecksOutputReference = exports.repositoryRulesetRulesRequiredStatusChecksToTerraform = exports.RepositoryRulesetRulesRequiredStatusChecksRequiredCheckList = exports.RepositoryRulesetRulesRequiredStatusChecksRequiredCheckOutputReference = exports.repositoryRulesetRulesRequiredStatusChecksRequiredCheckToTerraform = exports.RepositoryRulesetRulesRequiredDeploymentsOutputReference = exports.repositoryRulesetRulesRequiredDeploymentsToTerraform = exports.RepositoryRulesetRulesPullRequestOutputReference = exports.repositoryRulesetRulesPullRequestToTerraform = exports.RepositoryRulesetRulesCommitterEmailPatternOutputReference = exports.repositoryRulesetRulesCommitterEmailPatternToTerraform = exports.RepositoryRulesetRulesCommitMessagePatternOutputReference = exports.repositoryRulesetRulesCommitMessagePatternToTerraform = exports.RepositoryRulesetRulesCommitAuthorEmailPatternOutputReference = exports.repositoryRulesetRulesCommitAuthorEmailPatternToTerraform = exports.RepositoryRulesetRulesBranchNamePatternOutputReference = exports.repositoryRulesetRulesBranchNamePatternToTerraform = exports.RepositoryRulesetConditionsOutputReference = exports.repositoryRulesetConditionsToTerraform = exports.RepositoryRulesetConditionsRefNameOutputReference = exports.repositoryRulesetConditionsRefNameToTerraform = exports.RepositoryRulesetBypassActorsList = exports.RepositoryRulesetBypassActorsOutputReference = exports.repositoryRulesetBypassActorsToTerraform = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const cdktf = require("cdktf");
function repositoryRulesetBypassActorsToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        actor_id: cdktf.numberToTerraform(struct.actorId),
        actor_type: cdktf.stringToTerraform(struct.actorType),
        bypass_mode: cdktf.stringToTerraform(struct.bypassMode),
    };
}
exports.repositoryRulesetBypassActorsToTerraform = repositoryRulesetBypassActorsToTerraform;
class RepositoryRulesetBypassActorsOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    * @param complexObjectIndex the index of this item in the list
    * @param complexObjectIsFromSet whether the list is wrapping a set (will add tolist() to be able to access an item via an index)
    */
    constructor(terraformResource, terraformAttribute, complexObjectIndex, complexObjectIsFromSet) {
        super(terraformResource, terraformAttribute, complexObjectIsFromSet, complexObjectIndex);
        this.isEmptyObject = false;
    }
    get internalValue() {
        if (this.resolvableValue) {
            return this.resolvableValue;
        }
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._actorId !== undefined) {
            hasAnyValues = true;
            internalValueResult.actorId = this._actorId;
        }
        if (this._actorType !== undefined) {
            hasAnyValues = true;
            internalValueResult.actorType = this._actorType;
        }
        if (this._bypassMode !== undefined) {
            hasAnyValues = true;
            internalValueResult.bypassMode = this._bypassMode;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this.resolvableValue = undefined;
            this._actorId = undefined;
            this._actorType = undefined;
            this._bypassMode = undefined;
        }
        else if (cdktf.Tokenization.isResolvable(value)) {
            this.isEmptyObject = false;
            this.resolvableValue = value;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this.resolvableValue = undefined;
            this._actorId = value.actorId;
            this._actorType = value.actorType;
            this._bypassMode = value.bypassMode;
        }
    }
    get actorId() {
        return this.getNumberAttribute('actor_id');
    }
    set actorId(value) {
        this._actorId = value;
    }
    // Temporarily expose input value. Use with caution.
    get actorIdInput() {
        return this._actorId;
    }
    get actorType() {
        return this.getStringAttribute('actor_type');
    }
    set actorType(value) {
        this._actorType = value;
    }
    // Temporarily expose input value. Use with caution.
    get actorTypeInput() {
        return this._actorType;
    }
    get bypassMode() {
        return this.getStringAttribute('bypass_mode');
    }
    set bypassMode(value) {
        this._bypassMode = value;
    }
    // Temporarily expose input value. Use with caution.
    get bypassModeInput() {
        return this._bypassMode;
    }
}
exports.RepositoryRulesetBypassActorsOutputReference = RepositoryRulesetBypassActorsOutputReference;
_a = JSII_RTTI_SYMBOL_1;
RepositoryRulesetBypassActorsOutputReference[_a] = { fqn: "@cdktf/provider-github.repositoryRuleset.RepositoryRulesetBypassActorsOutputReference", version: "12.0.0" };
class RepositoryRulesetBypassActorsList extends cdktf.ComplexList {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    * @param wrapsSet whether the list is wrapping a set (will add tolist() to be able to access an item via an index)
    */
    constructor(terraformResource, terraformAttribute, wrapsSet) {
        super(terraformResource, terraformAttribute, wrapsSet);
        this.terraformResource = terraformResource;
        this.terraformAttribute = terraformAttribute;
        this.wrapsSet = wrapsSet;
    }
    /**
    * @param index the index of the item to return
    */
    get(index) {
        return new RepositoryRulesetBypassActorsOutputReference(this.terraformResource, this.terraformAttribute, index, this.wrapsSet);
    }
}
exports.RepositoryRulesetBypassActorsList = RepositoryRulesetBypassActorsList;
_b = JSII_RTTI_SYMBOL_1;
RepositoryRulesetBypassActorsList[_b] = { fqn: "@cdktf/provider-github.repositoryRuleset.RepositoryRulesetBypassActorsList", version: "12.0.0" };
function repositoryRulesetConditionsRefNameToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        exclude: cdktf.listMapper(cdktf.stringToTerraform, false)(struct.exclude),
        include: cdktf.listMapper(cdktf.stringToTerraform, false)(struct.include),
    };
}
exports.repositoryRulesetConditionsRefNameToTerraform = repositoryRulesetConditionsRefNameToTerraform;
class RepositoryRulesetConditionsRefNameOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._exclude !== undefined) {
            hasAnyValues = true;
            internalValueResult.exclude = this._exclude;
        }
        if (this._include !== undefined) {
            hasAnyValues = true;
            internalValueResult.include = this._include;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._exclude = undefined;
            this._include = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._exclude = value.exclude;
            this._include = value.include;
        }
    }
    get exclude() {
        return this.getListAttribute('exclude');
    }
    set exclude(value) {
        this._exclude = value;
    }
    // Temporarily expose input value. Use with caution.
    get excludeInput() {
        return this._exclude;
    }
    get include() {
        return this.getListAttribute('include');
    }
    set include(value) {
        this._include = value;
    }
    // Temporarily expose input value. Use with caution.
    get includeInput() {
        return this._include;
    }
}
exports.RepositoryRulesetConditionsRefNameOutputReference = RepositoryRulesetConditionsRefNameOutputReference;
_c = JSII_RTTI_SYMBOL_1;
RepositoryRulesetConditionsRefNameOutputReference[_c] = { fqn: "@cdktf/provider-github.repositoryRuleset.RepositoryRulesetConditionsRefNameOutputReference", version: "12.0.0" };
function repositoryRulesetConditionsToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        ref_name: repositoryRulesetConditionsRefNameToTerraform(struct.refName),
    };
}
exports.repositoryRulesetConditionsToTerraform = repositoryRulesetConditionsToTerraform;
class RepositoryRulesetConditionsOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
        // ref_name - computed: false, optional: false, required: true
        this._refName = new RepositoryRulesetConditionsRefNameOutputReference(this, "ref_name");
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._refName?.internalValue !== undefined) {
            hasAnyValues = true;
            internalValueResult.refName = this._refName?.internalValue;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._refName.internalValue = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._refName.internalValue = value.refName;
        }
    }
    get refName() {
        return this._refName;
    }
    putRefName(value) {
        this._refName.internalValue = value;
    }
    // Temporarily expose input value. Use with caution.
    get refNameInput() {
        return this._refName.internalValue;
    }
}
exports.RepositoryRulesetConditionsOutputReference = RepositoryRulesetConditionsOutputReference;
_d = JSII_RTTI_SYMBOL_1;
RepositoryRulesetConditionsOutputReference[_d] = { fqn: "@cdktf/provider-github.repositoryRuleset.RepositoryRulesetConditionsOutputReference", version: "12.0.0" };
function repositoryRulesetRulesBranchNamePatternToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        name: cdktf.stringToTerraform(struct.name),
        negate: cdktf.booleanToTerraform(struct.negate),
        operator: cdktf.stringToTerraform(struct.operator),
        pattern: cdktf.stringToTerraform(struct.pattern),
    };
}
exports.repositoryRulesetRulesBranchNamePatternToTerraform = repositoryRulesetRulesBranchNamePatternToTerraform;
class RepositoryRulesetRulesBranchNamePatternOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._name !== undefined) {
            hasAnyValues = true;
            internalValueResult.name = this._name;
        }
        if (this._negate !== undefined) {
            hasAnyValues = true;
            internalValueResult.negate = this._negate;
        }
        if (this._operator !== undefined) {
            hasAnyValues = true;
            internalValueResult.operator = this._operator;
        }
        if (this._pattern !== undefined) {
            hasAnyValues = true;
            internalValueResult.pattern = this._pattern;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._name = undefined;
            this._negate = undefined;
            this._operator = undefined;
            this._pattern = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._name = value.name;
            this._negate = value.negate;
            this._operator = value.operator;
            this._pattern = value.pattern;
        }
    }
    get name() {
        return this.getStringAttribute('name');
    }
    set name(value) {
        this._name = value;
    }
    resetName() {
        this._name = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get nameInput() {
        return this._name;
    }
    get negate() {
        return this.getBooleanAttribute('negate');
    }
    set negate(value) {
        this._negate = value;
    }
    resetNegate() {
        this._negate = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get negateInput() {
        return this._negate;
    }
    get operator() {
        return this.getStringAttribute('operator');
    }
    set operator(value) {
        this._operator = value;
    }
    // Temporarily expose input value. Use with caution.
    get operatorInput() {
        return this._operator;
    }
    get pattern() {
        return this.getStringAttribute('pattern');
    }
    set pattern(value) {
        this._pattern = value;
    }
    // Temporarily expose input value. Use with caution.
    get patternInput() {
        return this._pattern;
    }
}
exports.RepositoryRulesetRulesBranchNamePatternOutputReference = RepositoryRulesetRulesBranchNamePatternOutputReference;
_e = JSII_RTTI_SYMBOL_1;
RepositoryRulesetRulesBranchNamePatternOutputReference[_e] = { fqn: "@cdktf/provider-github.repositoryRuleset.RepositoryRulesetRulesBranchNamePatternOutputReference", version: "12.0.0" };
function repositoryRulesetRulesCommitAuthorEmailPatternToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        name: cdktf.stringToTerraform(struct.name),
        negate: cdktf.booleanToTerraform(struct.negate),
        operator: cdktf.stringToTerraform(struct.operator),
        pattern: cdktf.stringToTerraform(struct.pattern),
    };
}
exports.repositoryRulesetRulesCommitAuthorEmailPatternToTerraform = repositoryRulesetRulesCommitAuthorEmailPatternToTerraform;
class RepositoryRulesetRulesCommitAuthorEmailPatternOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._name !== undefined) {
            hasAnyValues = true;
            internalValueResult.name = this._name;
        }
        if (this._negate !== undefined) {
            hasAnyValues = true;
            internalValueResult.negate = this._negate;
        }
        if (this._operator !== undefined) {
            hasAnyValues = true;
            internalValueResult.operator = this._operator;
        }
        if (this._pattern !== undefined) {
            hasAnyValues = true;
            internalValueResult.pattern = this._pattern;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._name = undefined;
            this._negate = undefined;
            this._operator = undefined;
            this._pattern = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._name = value.name;
            this._negate = value.negate;
            this._operator = value.operator;
            this._pattern = value.pattern;
        }
    }
    get name() {
        return this.getStringAttribute('name');
    }
    set name(value) {
        this._name = value;
    }
    resetName() {
        this._name = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get nameInput() {
        return this._name;
    }
    get negate() {
        return this.getBooleanAttribute('negate');
    }
    set negate(value) {
        this._negate = value;
    }
    resetNegate() {
        this._negate = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get negateInput() {
        return this._negate;
    }
    get operator() {
        return this.getStringAttribute('operator');
    }
    set operator(value) {
        this._operator = value;
    }
    // Temporarily expose input value. Use with caution.
    get operatorInput() {
        return this._operator;
    }
    get pattern() {
        return this.getStringAttribute('pattern');
    }
    set pattern(value) {
        this._pattern = value;
    }
    // Temporarily expose input value. Use with caution.
    get patternInput() {
        return this._pattern;
    }
}
exports.RepositoryRulesetRulesCommitAuthorEmailPatternOutputReference = RepositoryRulesetRulesCommitAuthorEmailPatternOutputReference;
_f = JSII_RTTI_SYMBOL_1;
RepositoryRulesetRulesCommitAuthorEmailPatternOutputReference[_f] = { fqn: "@cdktf/provider-github.repositoryRuleset.RepositoryRulesetRulesCommitAuthorEmailPatternOutputReference", version: "12.0.0" };
function repositoryRulesetRulesCommitMessagePatternToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        name: cdktf.stringToTerraform(struct.name),
        negate: cdktf.booleanToTerraform(struct.negate),
        operator: cdktf.stringToTerraform(struct.operator),
        pattern: cdktf.stringToTerraform(struct.pattern),
    };
}
exports.repositoryRulesetRulesCommitMessagePatternToTerraform = repositoryRulesetRulesCommitMessagePatternToTerraform;
class RepositoryRulesetRulesCommitMessagePatternOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._name !== undefined) {
            hasAnyValues = true;
            internalValueResult.name = this._name;
        }
        if (this._negate !== undefined) {
            hasAnyValues = true;
            internalValueResult.negate = this._negate;
        }
        if (this._operator !== undefined) {
            hasAnyValues = true;
            internalValueResult.operator = this._operator;
        }
        if (this._pattern !== undefined) {
            hasAnyValues = true;
            internalValueResult.pattern = this._pattern;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._name = undefined;
            this._negate = undefined;
            this._operator = undefined;
            this._pattern = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._name = value.name;
            this._negate = value.negate;
            this._operator = value.operator;
            this._pattern = value.pattern;
        }
    }
    get name() {
        return this.getStringAttribute('name');
    }
    set name(value) {
        this._name = value;
    }
    resetName() {
        this._name = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get nameInput() {
        return this._name;
    }
    get negate() {
        return this.getBooleanAttribute('negate');
    }
    set negate(value) {
        this._negate = value;
    }
    resetNegate() {
        this._negate = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get negateInput() {
        return this._negate;
    }
    get operator() {
        return this.getStringAttribute('operator');
    }
    set operator(value) {
        this._operator = value;
    }
    // Temporarily expose input value. Use with caution.
    get operatorInput() {
        return this._operator;
    }
    get pattern() {
        return this.getStringAttribute('pattern');
    }
    set pattern(value) {
        this._pattern = value;
    }
    // Temporarily expose input value. Use with caution.
    get patternInput() {
        return this._pattern;
    }
}
exports.RepositoryRulesetRulesCommitMessagePatternOutputReference = RepositoryRulesetRulesCommitMessagePatternOutputReference;
_g = JSII_RTTI_SYMBOL_1;
RepositoryRulesetRulesCommitMessagePatternOutputReference[_g] = { fqn: "@cdktf/provider-github.repositoryRuleset.RepositoryRulesetRulesCommitMessagePatternOutputReference", version: "12.0.0" };
function repositoryRulesetRulesCommitterEmailPatternToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        name: cdktf.stringToTerraform(struct.name),
        negate: cdktf.booleanToTerraform(struct.negate),
        operator: cdktf.stringToTerraform(struct.operator),
        pattern: cdktf.stringToTerraform(struct.pattern),
    };
}
exports.repositoryRulesetRulesCommitterEmailPatternToTerraform = repositoryRulesetRulesCommitterEmailPatternToTerraform;
class RepositoryRulesetRulesCommitterEmailPatternOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._name !== undefined) {
            hasAnyValues = true;
            internalValueResult.name = this._name;
        }
        if (this._negate !== undefined) {
            hasAnyValues = true;
            internalValueResult.negate = this._negate;
        }
        if (this._operator !== undefined) {
            hasAnyValues = true;
            internalValueResult.operator = this._operator;
        }
        if (this._pattern !== undefined) {
            hasAnyValues = true;
            internalValueResult.pattern = this._pattern;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._name = undefined;
            this._negate = undefined;
            this._operator = undefined;
            this._pattern = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._name = value.name;
            this._negate = value.negate;
            this._operator = value.operator;
            this._pattern = value.pattern;
        }
    }
    get name() {
        return this.getStringAttribute('name');
    }
    set name(value) {
        this._name = value;
    }
    resetName() {
        this._name = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get nameInput() {
        return this._name;
    }
    get negate() {
        return this.getBooleanAttribute('negate');
    }
    set negate(value) {
        this._negate = value;
    }
    resetNegate() {
        this._negate = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get negateInput() {
        return this._negate;
    }
    get operator() {
        return this.getStringAttribute('operator');
    }
    set operator(value) {
        this._operator = value;
    }
    // Temporarily expose input value. Use with caution.
    get operatorInput() {
        return this._operator;
    }
    get pattern() {
        return this.getStringAttribute('pattern');
    }
    set pattern(value) {
        this._pattern = value;
    }
    // Temporarily expose input value. Use with caution.
    get patternInput() {
        return this._pattern;
    }
}
exports.RepositoryRulesetRulesCommitterEmailPatternOutputReference = RepositoryRulesetRulesCommitterEmailPatternOutputReference;
_h = JSII_RTTI_SYMBOL_1;
RepositoryRulesetRulesCommitterEmailPatternOutputReference[_h] = { fqn: "@cdktf/provider-github.repositoryRuleset.RepositoryRulesetRulesCommitterEmailPatternOutputReference", version: "12.0.0" };
function repositoryRulesetRulesPullRequestToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        dismiss_stale_reviews_on_push: cdktf.booleanToTerraform(struct.dismissStaleReviewsOnPush),
        require_code_owner_review: cdktf.booleanToTerraform(struct.requireCodeOwnerReview),
        require_last_push_approval: cdktf.booleanToTerraform(struct.requireLastPushApproval),
        required_approving_review_count: cdktf.numberToTerraform(struct.requiredApprovingReviewCount),
        required_review_thread_resolution: cdktf.booleanToTerraform(struct.requiredReviewThreadResolution),
    };
}
exports.repositoryRulesetRulesPullRequestToTerraform = repositoryRulesetRulesPullRequestToTerraform;
class RepositoryRulesetRulesPullRequestOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._dismissStaleReviewsOnPush !== undefined) {
            hasAnyValues = true;
            internalValueResult.dismissStaleReviewsOnPush = this._dismissStaleReviewsOnPush;
        }
        if (this._requireCodeOwnerReview !== undefined) {
            hasAnyValues = true;
            internalValueResult.requireCodeOwnerReview = this._requireCodeOwnerReview;
        }
        if (this._requireLastPushApproval !== undefined) {
            hasAnyValues = true;
            internalValueResult.requireLastPushApproval = this._requireLastPushApproval;
        }
        if (this._requiredApprovingReviewCount !== undefined) {
            hasAnyValues = true;
            internalValueResult.requiredApprovingReviewCount = this._requiredApprovingReviewCount;
        }
        if (this._requiredReviewThreadResolution !== undefined) {
            hasAnyValues = true;
            internalValueResult.requiredReviewThreadResolution = this._requiredReviewThreadResolution;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._dismissStaleReviewsOnPush = undefined;
            this._requireCodeOwnerReview = undefined;
            this._requireLastPushApproval = undefined;
            this._requiredApprovingReviewCount = undefined;
            this._requiredReviewThreadResolution = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._dismissStaleReviewsOnPush = value.dismissStaleReviewsOnPush;
            this._requireCodeOwnerReview = value.requireCodeOwnerReview;
            this._requireLastPushApproval = value.requireLastPushApproval;
            this._requiredApprovingReviewCount = value.requiredApprovingReviewCount;
            this._requiredReviewThreadResolution = value.requiredReviewThreadResolution;
        }
    }
    get dismissStaleReviewsOnPush() {
        return this.getBooleanAttribute('dismiss_stale_reviews_on_push');
    }
    set dismissStaleReviewsOnPush(value) {
        this._dismissStaleReviewsOnPush = value;
    }
    resetDismissStaleReviewsOnPush() {
        this._dismissStaleReviewsOnPush = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get dismissStaleReviewsOnPushInput() {
        return this._dismissStaleReviewsOnPush;
    }
    get requireCodeOwnerReview() {
        return this.getBooleanAttribute('require_code_owner_review');
    }
    set requireCodeOwnerReview(value) {
        this._requireCodeOwnerReview = value;
    }
    resetRequireCodeOwnerReview() {
        this._requireCodeOwnerReview = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get requireCodeOwnerReviewInput() {
        return this._requireCodeOwnerReview;
    }
    get requireLastPushApproval() {
        return this.getBooleanAttribute('require_last_push_approval');
    }
    set requireLastPushApproval(value) {
        this._requireLastPushApproval = value;
    }
    resetRequireLastPushApproval() {
        this._requireLastPushApproval = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get requireLastPushApprovalInput() {
        return this._requireLastPushApproval;
    }
    get requiredApprovingReviewCount() {
        return this.getNumberAttribute('required_approving_review_count');
    }
    set requiredApprovingReviewCount(value) {
        this._requiredApprovingReviewCount = value;
    }
    resetRequiredApprovingReviewCount() {
        this._requiredApprovingReviewCount = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get requiredApprovingReviewCountInput() {
        return this._requiredApprovingReviewCount;
    }
    get requiredReviewThreadResolution() {
        return this.getBooleanAttribute('required_review_thread_resolution');
    }
    set requiredReviewThreadResolution(value) {
        this._requiredReviewThreadResolution = value;
    }
    resetRequiredReviewThreadResolution() {
        this._requiredReviewThreadResolution = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get requiredReviewThreadResolutionInput() {
        return this._requiredReviewThreadResolution;
    }
}
exports.RepositoryRulesetRulesPullRequestOutputReference = RepositoryRulesetRulesPullRequestOutputReference;
_j = JSII_RTTI_SYMBOL_1;
RepositoryRulesetRulesPullRequestOutputReference[_j] = { fqn: "@cdktf/provider-github.repositoryRuleset.RepositoryRulesetRulesPullRequestOutputReference", version: "12.0.0" };
function repositoryRulesetRulesRequiredDeploymentsToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        required_deployment_environments: cdktf.listMapper(cdktf.stringToTerraform, false)(struct.requiredDeploymentEnvironments),
    };
}
exports.repositoryRulesetRulesRequiredDeploymentsToTerraform = repositoryRulesetRulesRequiredDeploymentsToTerraform;
class RepositoryRulesetRulesRequiredDeploymentsOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._requiredDeploymentEnvironments !== undefined) {
            hasAnyValues = true;
            internalValueResult.requiredDeploymentEnvironments = this._requiredDeploymentEnvironments;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._requiredDeploymentEnvironments = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._requiredDeploymentEnvironments = value.requiredDeploymentEnvironments;
        }
    }
    get requiredDeploymentEnvironments() {
        return this.getListAttribute('required_deployment_environments');
    }
    set requiredDeploymentEnvironments(value) {
        this._requiredDeploymentEnvironments = value;
    }
    resetRequiredDeploymentEnvironments() {
        this._requiredDeploymentEnvironments = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get requiredDeploymentEnvironmentsInput() {
        return this._requiredDeploymentEnvironments;
    }
}
exports.RepositoryRulesetRulesRequiredDeploymentsOutputReference = RepositoryRulesetRulesRequiredDeploymentsOutputReference;
_k = JSII_RTTI_SYMBOL_1;
RepositoryRulesetRulesRequiredDeploymentsOutputReference[_k] = { fqn: "@cdktf/provider-github.repositoryRuleset.RepositoryRulesetRulesRequiredDeploymentsOutputReference", version: "12.0.0" };
function repositoryRulesetRulesRequiredStatusChecksRequiredCheckToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        context: cdktf.stringToTerraform(struct.context),
        integration_id: cdktf.numberToTerraform(struct.integrationId),
    };
}
exports.repositoryRulesetRulesRequiredStatusChecksRequiredCheckToTerraform = repositoryRulesetRulesRequiredStatusChecksRequiredCheckToTerraform;
class RepositoryRulesetRulesRequiredStatusChecksRequiredCheckOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    * @param complexObjectIndex the index of this item in the list
    * @param complexObjectIsFromSet whether the list is wrapping a set (will add tolist() to be able to access an item via an index)
    */
    constructor(terraformResource, terraformAttribute, complexObjectIndex, complexObjectIsFromSet) {
        super(terraformResource, terraformAttribute, complexObjectIsFromSet, complexObjectIndex);
        this.isEmptyObject = false;
    }
    get internalValue() {
        if (this.resolvableValue) {
            return this.resolvableValue;
        }
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._context !== undefined) {
            hasAnyValues = true;
            internalValueResult.context = this._context;
        }
        if (this._integrationId !== undefined) {
            hasAnyValues = true;
            internalValueResult.integrationId = this._integrationId;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this.resolvableValue = undefined;
            this._context = undefined;
            this._integrationId = undefined;
        }
        else if (cdktf.Tokenization.isResolvable(value)) {
            this.isEmptyObject = false;
            this.resolvableValue = value;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this.resolvableValue = undefined;
            this._context = value.context;
            this._integrationId = value.integrationId;
        }
    }
    get context() {
        return this.getStringAttribute('context');
    }
    set context(value) {
        this._context = value;
    }
    // Temporarily expose input value. Use with caution.
    get contextInput() {
        return this._context;
    }
    get integrationId() {
        return this.getNumberAttribute('integration_id');
    }
    set integrationId(value) {
        this._integrationId = value;
    }
    resetIntegrationId() {
        this._integrationId = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get integrationIdInput() {
        return this._integrationId;
    }
}
exports.RepositoryRulesetRulesRequiredStatusChecksRequiredCheckOutputReference = RepositoryRulesetRulesRequiredStatusChecksRequiredCheckOutputReference;
_l = JSII_RTTI_SYMBOL_1;
RepositoryRulesetRulesRequiredStatusChecksRequiredCheckOutputReference[_l] = { fqn: "@cdktf/provider-github.repositoryRuleset.RepositoryRulesetRulesRequiredStatusChecksRequiredCheckOutputReference", version: "12.0.0" };
class RepositoryRulesetRulesRequiredStatusChecksRequiredCheckList extends cdktf.ComplexList {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    * @param wrapsSet whether the list is wrapping a set (will add tolist() to be able to access an item via an index)
    */
    constructor(terraformResource, terraformAttribute, wrapsSet) {
        super(terraformResource, terraformAttribute, wrapsSet);
        this.terraformResource = terraformResource;
        this.terraformAttribute = terraformAttribute;
        this.wrapsSet = wrapsSet;
    }
    /**
    * @param index the index of the item to return
    */
    get(index) {
        return new RepositoryRulesetRulesRequiredStatusChecksRequiredCheckOutputReference(this.terraformResource, this.terraformAttribute, index, this.wrapsSet);
    }
}
exports.RepositoryRulesetRulesRequiredStatusChecksRequiredCheckList = RepositoryRulesetRulesRequiredStatusChecksRequiredCheckList;
_m = JSII_RTTI_SYMBOL_1;
RepositoryRulesetRulesRequiredStatusChecksRequiredCheckList[_m] = { fqn: "@cdktf/provider-github.repositoryRuleset.RepositoryRulesetRulesRequiredStatusChecksRequiredCheckList", version: "12.0.0" };
function repositoryRulesetRulesRequiredStatusChecksToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        strict_required_status_checks_policy: cdktf.booleanToTerraform(struct.strictRequiredStatusChecksPolicy),
        required_check: cdktf.listMapper(repositoryRulesetRulesRequiredStatusChecksRequiredCheckToTerraform, true)(struct.requiredCheck),
    };
}
exports.repositoryRulesetRulesRequiredStatusChecksToTerraform = repositoryRulesetRulesRequiredStatusChecksToTerraform;
class RepositoryRulesetRulesRequiredStatusChecksOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
        // required_check - computed: false, optional: false, required: true
        this._requiredCheck = new RepositoryRulesetRulesRequiredStatusChecksRequiredCheckList(this, "required_check", true);
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._strictRequiredStatusChecksPolicy !== undefined) {
            hasAnyValues = true;
            internalValueResult.strictRequiredStatusChecksPolicy = this._strictRequiredStatusChecksPolicy;
        }
        if (this._requiredCheck?.internalValue !== undefined) {
            hasAnyValues = true;
            internalValueResult.requiredCheck = this._requiredCheck?.internalValue;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._strictRequiredStatusChecksPolicy = undefined;
            this._requiredCheck.internalValue = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._strictRequiredStatusChecksPolicy = value.strictRequiredStatusChecksPolicy;
            this._requiredCheck.internalValue = value.requiredCheck;
        }
    }
    get strictRequiredStatusChecksPolicy() {
        return this.getBooleanAttribute('strict_required_status_checks_policy');
    }
    set strictRequiredStatusChecksPolicy(value) {
        this._strictRequiredStatusChecksPolicy = value;
    }
    resetStrictRequiredStatusChecksPolicy() {
        this._strictRequiredStatusChecksPolicy = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get strictRequiredStatusChecksPolicyInput() {
        return this._strictRequiredStatusChecksPolicy;
    }
    get requiredCheck() {
        return this._requiredCheck;
    }
    putRequiredCheck(value) {
        this._requiredCheck.internalValue = value;
    }
    // Temporarily expose input value. Use with caution.
    get requiredCheckInput() {
        return this._requiredCheck.internalValue;
    }
}
exports.RepositoryRulesetRulesRequiredStatusChecksOutputReference = RepositoryRulesetRulesRequiredStatusChecksOutputReference;
_o = JSII_RTTI_SYMBOL_1;
RepositoryRulesetRulesRequiredStatusChecksOutputReference[_o] = { fqn: "@cdktf/provider-github.repositoryRuleset.RepositoryRulesetRulesRequiredStatusChecksOutputReference", version: "12.0.0" };
function repositoryRulesetRulesTagNamePatternToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        name: cdktf.stringToTerraform(struct.name),
        negate: cdktf.booleanToTerraform(struct.negate),
        operator: cdktf.stringToTerraform(struct.operator),
        pattern: cdktf.stringToTerraform(struct.pattern),
    };
}
exports.repositoryRulesetRulesTagNamePatternToTerraform = repositoryRulesetRulesTagNamePatternToTerraform;
class RepositoryRulesetRulesTagNamePatternOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._name !== undefined) {
            hasAnyValues = true;
            internalValueResult.name = this._name;
        }
        if (this._negate !== undefined) {
            hasAnyValues = true;
            internalValueResult.negate = this._negate;
        }
        if (this._operator !== undefined) {
            hasAnyValues = true;
            internalValueResult.operator = this._operator;
        }
        if (this._pattern !== undefined) {
            hasAnyValues = true;
            internalValueResult.pattern = this._pattern;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._name = undefined;
            this._negate = undefined;
            this._operator = undefined;
            this._pattern = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._name = value.name;
            this._negate = value.negate;
            this._operator = value.operator;
            this._pattern = value.pattern;
        }
    }
    get name() {
        return this.getStringAttribute('name');
    }
    set name(value) {
        this._name = value;
    }
    resetName() {
        this._name = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get nameInput() {
        return this._name;
    }
    get negate() {
        return this.getBooleanAttribute('negate');
    }
    set negate(value) {
        this._negate = value;
    }
    resetNegate() {
        this._negate = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get negateInput() {
        return this._negate;
    }
    get operator() {
        return this.getStringAttribute('operator');
    }
    set operator(value) {
        this._operator = value;
    }
    // Temporarily expose input value. Use with caution.
    get operatorInput() {
        return this._operator;
    }
    get pattern() {
        return this.getStringAttribute('pattern');
    }
    set pattern(value) {
        this._pattern = value;
    }
    // Temporarily expose input value. Use with caution.
    get patternInput() {
        return this._pattern;
    }
}
exports.RepositoryRulesetRulesTagNamePatternOutputReference = RepositoryRulesetRulesTagNamePatternOutputReference;
_p = JSII_RTTI_SYMBOL_1;
RepositoryRulesetRulesTagNamePatternOutputReference[_p] = { fqn: "@cdktf/provider-github.repositoryRuleset.RepositoryRulesetRulesTagNamePatternOutputReference", version: "12.0.0" };
function repositoryRulesetRulesToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        creation: cdktf.booleanToTerraform(struct.creation),
        deletion: cdktf.booleanToTerraform(struct.deletion),
        non_fast_forward: cdktf.booleanToTerraform(struct.nonFastForward),
        required_linear_history: cdktf.booleanToTerraform(struct.requiredLinearHistory),
        required_signatures: cdktf.booleanToTerraform(struct.requiredSignatures),
        update: cdktf.booleanToTerraform(struct.update),
        update_allows_fetch_and_merge: cdktf.booleanToTerraform(struct.updateAllowsFetchAndMerge),
        branch_name_pattern: repositoryRulesetRulesBranchNamePatternToTerraform(struct.branchNamePattern),
        commit_author_email_pattern: repositoryRulesetRulesCommitAuthorEmailPatternToTerraform(struct.commitAuthorEmailPattern),
        commit_message_pattern: repositoryRulesetRulesCommitMessagePatternToTerraform(struct.commitMessagePattern),
        committer_email_pattern: repositoryRulesetRulesCommitterEmailPatternToTerraform(struct.committerEmailPattern),
        pull_request: repositoryRulesetRulesPullRequestToTerraform(struct.pullRequest),
        required_deployments: repositoryRulesetRulesRequiredDeploymentsToTerraform(struct.requiredDeployments),
        required_status_checks: repositoryRulesetRulesRequiredStatusChecksToTerraform(struct.requiredStatusChecks),
        tag_name_pattern: repositoryRulesetRulesTagNamePatternToTerraform(struct.tagNamePattern),
    };
}
exports.repositoryRulesetRulesToTerraform = repositoryRulesetRulesToTerraform;
class RepositoryRulesetRulesOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
        // branch_name_pattern - computed: false, optional: true, required: false
        this._branchNamePattern = new RepositoryRulesetRulesBranchNamePatternOutputReference(this, "branch_name_pattern");
        // commit_author_email_pattern - computed: false, optional: true, required: false
        this._commitAuthorEmailPattern = new RepositoryRulesetRulesCommitAuthorEmailPatternOutputReference(this, "commit_author_email_pattern");
        // commit_message_pattern - computed: false, optional: true, required: false
        this._commitMessagePattern = new RepositoryRulesetRulesCommitMessagePatternOutputReference(this, "commit_message_pattern");
        // committer_email_pattern - computed: false, optional: true, required: false
        this._committerEmailPattern = new RepositoryRulesetRulesCommitterEmailPatternOutputReference(this, "committer_email_pattern");
        // pull_request - computed: false, optional: true, required: false
        this._pullRequest = new RepositoryRulesetRulesPullRequestOutputReference(this, "pull_request");
        // required_deployments - computed: false, optional: true, required: false
        this._requiredDeployments = new RepositoryRulesetRulesRequiredDeploymentsOutputReference(this, "required_deployments");
        // required_status_checks - computed: false, optional: true, required: false
        this._requiredStatusChecks = new RepositoryRulesetRulesRequiredStatusChecksOutputReference(this, "required_status_checks");
        // tag_name_pattern - computed: false, optional: true, required: false
        this._tagNamePattern = new RepositoryRulesetRulesTagNamePatternOutputReference(this, "tag_name_pattern");
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._creation !== undefined) {
            hasAnyValues = true;
            internalValueResult.creation = this._creation;
        }
        if (this._deletion !== undefined) {
            hasAnyValues = true;
            internalValueResult.deletion = this._deletion;
        }
        if (this._nonFastForward !== undefined) {
            hasAnyValues = true;
            internalValueResult.nonFastForward = this._nonFastForward;
        }
        if (this._requiredLinearHistory !== undefined) {
            hasAnyValues = true;
            internalValueResult.requiredLinearHistory = this._requiredLinearHistory;
        }
        if (this._requiredSignatures !== undefined) {
            hasAnyValues = true;
            internalValueResult.requiredSignatures = this._requiredSignatures;
        }
        if (this._update !== undefined) {
            hasAnyValues = true;
            internalValueResult.update = this._update;
        }
        if (this._updateAllowsFetchAndMerge !== undefined) {
            hasAnyValues = true;
            internalValueResult.updateAllowsFetchAndMerge = this._updateAllowsFetchAndMerge;
        }
        if (this._branchNamePattern?.internalValue !== undefined) {
            hasAnyValues = true;
            internalValueResult.branchNamePattern = this._branchNamePattern?.internalValue;
        }
        if (this._commitAuthorEmailPattern?.internalValue !== undefined) {
            hasAnyValues = true;
            internalValueResult.commitAuthorEmailPattern = this._commitAuthorEmailPattern?.internalValue;
        }
        if (this._commitMessagePattern?.internalValue !== undefined) {
            hasAnyValues = true;
            internalValueResult.commitMessagePattern = this._commitMessagePattern?.internalValue;
        }
        if (this._committerEmailPattern?.internalValue !== undefined) {
            hasAnyValues = true;
            internalValueResult.committerEmailPattern = this._committerEmailPattern?.internalValue;
        }
        if (this._pullRequest?.internalValue !== undefined) {
            hasAnyValues = true;
            internalValueResult.pullRequest = this._pullRequest?.internalValue;
        }
        if (this._requiredDeployments?.internalValue !== undefined) {
            hasAnyValues = true;
            internalValueResult.requiredDeployments = this._requiredDeployments?.internalValue;
        }
        if (this._requiredStatusChecks?.internalValue !== undefined) {
            hasAnyValues = true;
            internalValueResult.requiredStatusChecks = this._requiredStatusChecks?.internalValue;
        }
        if (this._tagNamePattern?.internalValue !== undefined) {
            hasAnyValues = true;
            internalValueResult.tagNamePattern = this._tagNamePattern?.internalValue;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._creation = undefined;
            this._deletion = undefined;
            this._nonFastForward = undefined;
            this._requiredLinearHistory = undefined;
            this._requiredSignatures = undefined;
            this._update = undefined;
            this._updateAllowsFetchAndMerge = undefined;
            this._branchNamePattern.internalValue = undefined;
            this._commitAuthorEmailPattern.internalValue = undefined;
            this._commitMessagePattern.internalValue = undefined;
            this._committerEmailPattern.internalValue = undefined;
            this._pullRequest.internalValue = undefined;
            this._requiredDeployments.internalValue = undefined;
            this._requiredStatusChecks.internalValue = undefined;
            this._tagNamePattern.internalValue = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._creation = value.creation;
            this._deletion = value.deletion;
            this._nonFastForward = value.nonFastForward;
            this._requiredLinearHistory = value.requiredLinearHistory;
            this._requiredSignatures = value.requiredSignatures;
            this._update = value.update;
            this._updateAllowsFetchAndMerge = value.updateAllowsFetchAndMerge;
            this._branchNamePattern.internalValue = value.branchNamePattern;
            this._commitAuthorEmailPattern.internalValue = value.commitAuthorEmailPattern;
            this._commitMessagePattern.internalValue = value.commitMessagePattern;
            this._committerEmailPattern.internalValue = value.committerEmailPattern;
            this._pullRequest.internalValue = value.pullRequest;
            this._requiredDeployments.internalValue = value.requiredDeployments;
            this._requiredStatusChecks.internalValue = value.requiredStatusChecks;
            this._tagNamePattern.internalValue = value.tagNamePattern;
        }
    }
    get creation() {
        return this.getBooleanAttribute('creation');
    }
    set creation(value) {
        this._creation = value;
    }
    resetCreation() {
        this._creation = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get creationInput() {
        return this._creation;
    }
    get deletion() {
        return this.getBooleanAttribute('deletion');
    }
    set deletion(value) {
        this._deletion = value;
    }
    resetDeletion() {
        this._deletion = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get deletionInput() {
        return this._deletion;
    }
    get nonFastForward() {
        return this.getBooleanAttribute('non_fast_forward');
    }
    set nonFastForward(value) {
        this._nonFastForward = value;
    }
    resetNonFastForward() {
        this._nonFastForward = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get nonFastForwardInput() {
        return this._nonFastForward;
    }
    get requiredLinearHistory() {
        return this.getBooleanAttribute('required_linear_history');
    }
    set requiredLinearHistory(value) {
        this._requiredLinearHistory = value;
    }
    resetRequiredLinearHistory() {
        this._requiredLinearHistory = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get requiredLinearHistoryInput() {
        return this._requiredLinearHistory;
    }
    get requiredSignatures() {
        return this.getBooleanAttribute('required_signatures');
    }
    set requiredSignatures(value) {
        this._requiredSignatures = value;
    }
    resetRequiredSignatures() {
        this._requiredSignatures = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get requiredSignaturesInput() {
        return this._requiredSignatures;
    }
    get update() {
        return this.getBooleanAttribute('update');
    }
    set update(value) {
        this._update = value;
    }
    resetUpdate() {
        this._update = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get updateInput() {
        return this._update;
    }
    get updateAllowsFetchAndMerge() {
        return this.getBooleanAttribute('update_allows_fetch_and_merge');
    }
    set updateAllowsFetchAndMerge(value) {
        this._updateAllowsFetchAndMerge = value;
    }
    resetUpdateAllowsFetchAndMerge() {
        this._updateAllowsFetchAndMerge = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get updateAllowsFetchAndMergeInput() {
        return this._updateAllowsFetchAndMerge;
    }
    get branchNamePattern() {
        return this._branchNamePattern;
    }
    putBranchNamePattern(value) {
        this._branchNamePattern.internalValue = value;
    }
    resetBranchNamePattern() {
        this._branchNamePattern.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get branchNamePatternInput() {
        return this._branchNamePattern.internalValue;
    }
    get commitAuthorEmailPattern() {
        return this._commitAuthorEmailPattern;
    }
    putCommitAuthorEmailPattern(value) {
        this._commitAuthorEmailPattern.internalValue = value;
    }
    resetCommitAuthorEmailPattern() {
        this._commitAuthorEmailPattern.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get commitAuthorEmailPatternInput() {
        return this._commitAuthorEmailPattern.internalValue;
    }
    get commitMessagePattern() {
        return this._commitMessagePattern;
    }
    putCommitMessagePattern(value) {
        this._commitMessagePattern.internalValue = value;
    }
    resetCommitMessagePattern() {
        this._commitMessagePattern.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get commitMessagePatternInput() {
        return this._commitMessagePattern.internalValue;
    }
    get committerEmailPattern() {
        return this._committerEmailPattern;
    }
    putCommitterEmailPattern(value) {
        this._committerEmailPattern.internalValue = value;
    }
    resetCommitterEmailPattern() {
        this._committerEmailPattern.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get committerEmailPatternInput() {
        return this._committerEmailPattern.internalValue;
    }
    get pullRequest() {
        return this._pullRequest;
    }
    putPullRequest(value) {
        this._pullRequest.internalValue = value;
    }
    resetPullRequest() {
        this._pullRequest.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get pullRequestInput() {
        return this._pullRequest.internalValue;
    }
    get requiredDeployments() {
        return this._requiredDeployments;
    }
    putRequiredDeployments(value) {
        this._requiredDeployments.internalValue = value;
    }
    resetRequiredDeployments() {
        this._requiredDeployments.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get requiredDeploymentsInput() {
        return this._requiredDeployments.internalValue;
    }
    get requiredStatusChecks() {
        return this._requiredStatusChecks;
    }
    putRequiredStatusChecks(value) {
        this._requiredStatusChecks.internalValue = value;
    }
    resetRequiredStatusChecks() {
        this._requiredStatusChecks.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get requiredStatusChecksInput() {
        return this._requiredStatusChecks.internalValue;
    }
    get tagNamePattern() {
        return this._tagNamePattern;
    }
    putTagNamePattern(value) {
        this._tagNamePattern.internalValue = value;
    }
    resetTagNamePattern() {
        this._tagNamePattern.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get tagNamePatternInput() {
        return this._tagNamePattern.internalValue;
    }
}
exports.RepositoryRulesetRulesOutputReference = RepositoryRulesetRulesOutputReference;
_q = JSII_RTTI_SYMBOL_1;
RepositoryRulesetRulesOutputReference[_q] = { fqn: "@cdktf/provider-github.repositoryRuleset.RepositoryRulesetRulesOutputReference", version: "12.0.0" };
/**
* Represents a {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/repository_ruleset github_repository_ruleset}
*/
class RepositoryRuleset extends cdktf.TerraformResource {
    // ==============
    // STATIC Methods
    // ==============
    /**
    * Generates CDKTF code for importing a RepositoryRuleset resource upon running "cdktf plan <stack-name>"
    * @param scope The scope in which to define this construct
    * @param importToId The construct id used in the generated config for the RepositoryRuleset to import
    * @param importFromId The id of the existing RepositoryRuleset that should be imported. Refer to the {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/repository_ruleset#import import section} in the documentation of this resource for the id to use
    * @param provider? Optional instance of the provider where the RepositoryRuleset to import is found
    */
    static generateConfigForImport(scope, importToId, importFromId, provider) {
        return new cdktf.ImportableResource(scope, importToId, { terraformResourceType: "github_repository_ruleset", importId: importFromId, provider });
    }
    // ===========
    // INITIALIZER
    // ===========
    /**
    * Create a new {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/repository_ruleset github_repository_ruleset} Resource
    *
    * @param scope The scope in which to define this construct
    * @param id The scoped construct ID. Must be unique amongst siblings in the same scope
    * @param options RepositoryRulesetConfig
    */
    constructor(scope, id, config) {
        super(scope, id, {
            terraformResourceType: 'github_repository_ruleset',
            terraformGeneratorMetadata: {
                providerName: 'github',
                providerVersion: '5.40.0',
                providerVersionConstraint: '~> 5.0'
            },
            provider: config.provider,
            dependsOn: config.dependsOn,
            count: config.count,
            lifecycle: config.lifecycle,
            provisioners: config.provisioners,
            connection: config.connection,
            forEach: config.forEach
        });
        // bypass_actors - computed: false, optional: true, required: false
        this._bypassActors = new RepositoryRulesetBypassActorsList(this, "bypass_actors", false);
        // conditions - computed: false, optional: true, required: false
        this._conditions = new RepositoryRulesetConditionsOutputReference(this, "conditions");
        // rules - computed: false, optional: false, required: true
        this._rules = new RepositoryRulesetRulesOutputReference(this, "rules");
        this._enforcement = config.enforcement;
        this._id = config.id;
        this._name = config.name;
        this._repository = config.repository;
        this._target = config.target;
        this._bypassActors.internalValue = config.bypassActors;
        this._conditions.internalValue = config.conditions;
        this._rules.internalValue = config.rules;
    }
    get enforcement() {
        return this.getStringAttribute('enforcement');
    }
    set enforcement(value) {
        this._enforcement = value;
    }
    // Temporarily expose input value. Use with caution.
    get enforcementInput() {
        return this._enforcement;
    }
    // etag - computed: true, optional: false, required: false
    get etag() {
        return this.getStringAttribute('etag');
    }
    get id() {
        return this.getStringAttribute('id');
    }
    set id(value) {
        this._id = value;
    }
    resetId() {
        this._id = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get idInput() {
        return this._id;
    }
    get name() {
        return this.getStringAttribute('name');
    }
    set name(value) {
        this._name = value;
    }
    // Temporarily expose input value. Use with caution.
    get nameInput() {
        return this._name;
    }
    // node_id - computed: true, optional: false, required: false
    get nodeId() {
        return this.getStringAttribute('node_id');
    }
    get repository() {
        return this.getStringAttribute('repository');
    }
    set repository(value) {
        this._repository = value;
    }
    resetRepository() {
        this._repository = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get repositoryInput() {
        return this._repository;
    }
    // ruleset_id - computed: true, optional: false, required: false
    get rulesetId() {
        return this.getNumberAttribute('ruleset_id');
    }
    get target() {
        return this.getStringAttribute('target');
    }
    set target(value) {
        this._target = value;
    }
    // Temporarily expose input value. Use with caution.
    get targetInput() {
        return this._target;
    }
    get bypassActors() {
        return this._bypassActors;
    }
    putBypassActors(value) {
        this._bypassActors.internalValue = value;
    }
    resetBypassActors() {
        this._bypassActors.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get bypassActorsInput() {
        return this._bypassActors.internalValue;
    }
    get conditions() {
        return this._conditions;
    }
    putConditions(value) {
        this._conditions.internalValue = value;
    }
    resetConditions() {
        this._conditions.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get conditionsInput() {
        return this._conditions.internalValue;
    }
    get rules() {
        return this._rules;
    }
    putRules(value) {
        this._rules.internalValue = value;
    }
    // Temporarily expose input value. Use with caution.
    get rulesInput() {
        return this._rules.internalValue;
    }
    // =========
    // SYNTHESIS
    // =========
    synthesizeAttributes() {
        return {
            enforcement: cdktf.stringToTerraform(this._enforcement),
            id: cdktf.stringToTerraform(this._id),
            name: cdktf.stringToTerraform(this._name),
            repository: cdktf.stringToTerraform(this._repository),
            target: cdktf.stringToTerraform(this._target),
            bypass_actors: cdktf.listMapper(repositoryRulesetBypassActorsToTerraform, true)(this._bypassActors.internalValue),
            conditions: repositoryRulesetConditionsToTerraform(this._conditions.internalValue),
            rules: repositoryRulesetRulesToTerraform(this._rules.internalValue),
        };
    }
}
exports.RepositoryRuleset = RepositoryRuleset;
_r = JSII_RTTI_SYMBOL_1;
RepositoryRuleset[_r] = { fqn: "@cdktf/provider-github.repositoryRuleset.RepositoryRuleset", version: "12.0.0" };
// =================
// STATIC PROPERTIES
// =================
RepositoryRuleset.tfResourceType = "github_repository_ruleset";
