// @cdktf/provider-github/lib/repository-environment/index.js
"use strict";
var _a, _b, _c, _d;
Object.defineProperty(exports, "__esModule", { value: true });
exports.RepositoryEnvironment = exports.RepositoryEnvironmentReviewersList = exports.RepositoryEnvironmentReviewersOutputReference = exports.repositoryEnvironmentReviewersToTerraform = exports.RepositoryEnvironmentDeploymentBranchPolicyOutputReference = exports.repositoryEnvironmentDeploymentBranchPolicyToTerraform = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const cdktf = require("cdktf");
function repositoryEnvironmentDeploymentBranchPolicyToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        custom_branch_policies: cdktf.booleanToTerraform(struct.customBranchPolicies),
        protected_branches: cdktf.booleanToTerraform(struct.protectedBranches),
    };
}
exports.repositoryEnvironmentDeploymentBranchPolicyToTerraform = repositoryEnvironmentDeploymentBranchPolicyToTerraform;
class RepositoryEnvironmentDeploymentBranchPolicyOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._customBranchPolicies !== undefined) {
            hasAnyValues = true;
            internalValueResult.customBranchPolicies = this._customBranchPolicies;
        }
        if (this._protectedBranches !== undefined) {
            hasAnyValues = true;
            internalValueResult.protectedBranches = this._protectedBranches;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._customBranchPolicies = undefined;
            this._protectedBranches = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._customBranchPolicies = value.customBranchPolicies;
            this._protectedBranches = value.protectedBranches;
        }
    }
    get customBranchPolicies() {
        return this.getBooleanAttribute('custom_branch_policies');
    }
    set customBranchPolicies(value) {
        this._customBranchPolicies = value;
    }
    // Temporarily expose input value. Use with caution.
    get customBranchPoliciesInput() {
        return this._customBranchPolicies;
    }
    get protectedBranches() {
        return this.getBooleanAttribute('protected_branches');
    }
    set protectedBranches(value) {
        this._protectedBranches = value;
    }
    // Temporarily expose input value. Use with caution.
    get protectedBranchesInput() {
        return this._protectedBranches;
    }
}
exports.RepositoryEnvironmentDeploymentBranchPolicyOutputReference = RepositoryEnvironmentDeploymentBranchPolicyOutputReference;
_a = JSII_RTTI_SYMBOL_1;
RepositoryEnvironmentDeploymentBranchPolicyOutputReference[_a] = { fqn: "@cdktf/provider-github.repositoryEnvironment.RepositoryEnvironmentDeploymentBranchPolicyOutputReference", version: "12.0.0" };
function repositoryEnvironmentReviewersToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        teams: cdktf.listMapper(cdktf.numberToTerraform, false)(struct.teams),
        users: cdktf.listMapper(cdktf.numberToTerraform, false)(struct.users),
    };
}
exports.repositoryEnvironmentReviewersToTerraform = repositoryEnvironmentReviewersToTerraform;
class RepositoryEnvironmentReviewersOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    * @param complexObjectIndex the index of this item in the list
    * @param complexObjectIsFromSet whether the list is wrapping a set (will add tolist() to be able to access an item via an index)
    */
    constructor(terraformResource, terraformAttribute, complexObjectIndex, complexObjectIsFromSet) {
        super(terraformResource, terraformAttribute, complexObjectIsFromSet, complexObjectIndex);
        this.isEmptyObject = false;
    }
    get internalValue() {
        if (this.resolvableValue) {
            return this.resolvableValue;
        }
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._teams !== undefined) {
            hasAnyValues = true;
            internalValueResult.teams = this._teams;
        }
        if (this._users !== undefined) {
            hasAnyValues = true;
            internalValueResult.users = this._users;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this.resolvableValue = undefined;
            this._teams = undefined;
            this._users = undefined;
        }
        else if (cdktf.Tokenization.isResolvable(value)) {
            this.isEmptyObject = false;
            this.resolvableValue = value;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this.resolvableValue = undefined;
            this._teams = value.teams;
            this._users = value.users;
        }
    }
    get teams() {
        return cdktf.Token.asNumberList(cdktf.Fn.tolist(this.getNumberListAttribute('teams')));
    }
    set teams(value) {
        this._teams = value;
    }
    resetTeams() {
        this._teams = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get teamsInput() {
        return this._teams;
    }
    get users() {
        return cdktf.Token.asNumberList(cdktf.Fn.tolist(this.getNumberListAttribute('users')));
    }
    set users(value) {
        this._users = value;
    }
    resetUsers() {
        this._users = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get usersInput() {
        return this._users;
    }
}
exports.RepositoryEnvironmentReviewersOutputReference = RepositoryEnvironmentReviewersOutputReference;
_b = JSII_RTTI_SYMBOL_1;
RepositoryEnvironmentReviewersOutputReference[_b] = { fqn: "@cdktf/provider-github.repositoryEnvironment.RepositoryEnvironmentReviewersOutputReference", version: "12.0.0" };
class RepositoryEnvironmentReviewersList extends cdktf.ComplexList {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    * @param wrapsSet whether the list is wrapping a set (will add tolist() to be able to access an item via an index)
    */
    constructor(terraformResource, terraformAttribute, wrapsSet) {
        super(terraformResource, terraformAttribute, wrapsSet);
        this.terraformResource = terraformResource;
        this.terraformAttribute = terraformAttribute;
        this.wrapsSet = wrapsSet;
    }
    /**
    * @param index the index of the item to return
    */
    get(index) {
        return new RepositoryEnvironmentReviewersOutputReference(this.terraformResource, this.terraformAttribute, index, this.wrapsSet);
    }
}
exports.RepositoryEnvironmentReviewersList = RepositoryEnvironmentReviewersList;
_c = JSII_RTTI_SYMBOL_1;
RepositoryEnvironmentReviewersList[_c] = { fqn: "@cdktf/provider-github.repositoryEnvironment.RepositoryEnvironmentReviewersList", version: "12.0.0" };
/**
* Represents a {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/repository_environment github_repository_environment}
*/
class RepositoryEnvironment extends cdktf.TerraformResource {
    // ==============
    // STATIC Methods
    // ==============
    /**
    * Generates CDKTF code for importing a RepositoryEnvironment resource upon running "cdktf plan <stack-name>"
    * @param scope The scope in which to define this construct
    * @param importToId The construct id used in the generated config for the RepositoryEnvironment to import
    * @param importFromId The id of the existing RepositoryEnvironment that should be imported. Refer to the {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/repository_environment#import import section} in the documentation of this resource for the id to use
    * @param provider? Optional instance of the provider where the RepositoryEnvironment to import is found
    */
    static generateConfigForImport(scope, importToId, importFromId, provider) {
        return new cdktf.ImportableResource(scope, importToId, { terraformResourceType: "github_repository_environment", importId: importFromId, provider });
    }
    // ===========
    // INITIALIZER
    // ===========
    /**
    * Create a new {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/repository_environment github_repository_environment} Resource
    *
    * @param scope The scope in which to define this construct
    * @param id The scoped construct ID. Must be unique amongst siblings in the same scope
    * @param options RepositoryEnvironmentConfig
    */
    constructor(scope, id, config) {
        super(scope, id, {
            terraformResourceType: 'github_repository_environment',
            terraformGeneratorMetadata: {
                providerName: 'github',
                providerVersion: '5.40.0',
                providerVersionConstraint: '~> 5.0'
            },
            provider: config.provider,
            dependsOn: config.dependsOn,
            count: config.count,
            lifecycle: config.lifecycle,
            provisioners: config.provisioners,
            connection: config.connection,
            forEach: config.forEach
        });
        // deployment_branch_policy - computed: false, optional: true, required: false
        this._deploymentBranchPolicy = new RepositoryEnvironmentDeploymentBranchPolicyOutputReference(this, "deployment_branch_policy");
        // reviewers - computed: false, optional: true, required: false
        this._reviewers = new RepositoryEnvironmentReviewersList(this, "reviewers", false);
        this._environment = config.environment;
        this._id = config.id;
        this._repository = config.repository;
        this._waitTimer = config.waitTimer;
        this._deploymentBranchPolicy.internalValue = config.deploymentBranchPolicy;
        this._reviewers.internalValue = config.reviewers;
    }
    get environment() {
        return this.getStringAttribute('environment');
    }
    set environment(value) {
        this._environment = value;
    }
    // Temporarily expose input value. Use with caution.
    get environmentInput() {
        return this._environment;
    }
    get id() {
        return this.getStringAttribute('id');
    }
    set id(value) {
        this._id = value;
    }
    resetId() {
        this._id = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get idInput() {
        return this._id;
    }
    get repository() {
        return this.getStringAttribute('repository');
    }
    set repository(value) {
        this._repository = value;
    }
    // Temporarily expose input value. Use with caution.
    get repositoryInput() {
        return this._repository;
    }
    get waitTimer() {
        return this.getNumberAttribute('wait_timer');
    }
    set waitTimer(value) {
        this._waitTimer = value;
    }
    resetWaitTimer() {
        this._waitTimer = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get waitTimerInput() {
        return this._waitTimer;
    }
    get deploymentBranchPolicy() {
        return this._deploymentBranchPolicy;
    }
    putDeploymentBranchPolicy(value) {
        this._deploymentBranchPolicy.internalValue = value;
    }
    resetDeploymentBranchPolicy() {
        this._deploymentBranchPolicy.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get deploymentBranchPolicyInput() {
        return this._deploymentBranchPolicy.internalValue;
    }
    get reviewers() {
        return this._reviewers;
    }
    putReviewers(value) {
        this._reviewers.internalValue = value;
    }
    resetReviewers() {
        this._reviewers.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get reviewersInput() {
        return this._reviewers.internalValue;
    }
    // =========
    // SYNTHESIS
    // =========
    synthesizeAttributes() {
        return {
            environment: cdktf.stringToTerraform(this._environment),
            id: cdktf.stringToTerraform(this._id),
            repository: cdktf.stringToTerraform(this._repository),
            wait_timer: cdktf.numberToTerraform(this._waitTimer),
            deployment_branch_policy: repositoryEnvironmentDeploymentBranchPolicyToTerraform(this._deploymentBranchPolicy.internalValue),
            reviewers: cdktf.listMapper(repositoryEnvironmentReviewersToTerraform, true)(this._reviewers.internalValue),
        };
    }
}
exports.RepositoryEnvironment = RepositoryEnvironment;
_d = JSII_RTTI_SYMBOL_1;
RepositoryEnvironment[_d] = { fqn: "@cdktf/provider-github.repositoryEnvironment.RepositoryEnvironment", version: "12.0.0" };
// =================
// STATIC PROPERTIES
// =================
RepositoryEnvironment.tfResourceType = "github_repository_environment";
