// @cdktf/provider-github/lib/data-github-repository/index.js
"use strict";
var _a, _b, _c, _d, _e, _f, _g;
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataGithubRepository = exports.DataGithubRepositoryTemplateList = exports.DataGithubRepositoryTemplateOutputReference = exports.dataGithubRepositoryTemplateToTerraform = exports.DataGithubRepositoryPagesList = exports.DataGithubRepositoryPagesOutputReference = exports.dataGithubRepositoryPagesToTerraform = exports.DataGithubRepositoryPagesSourceList = exports.DataGithubRepositoryPagesSourceOutputReference = exports.dataGithubRepositoryPagesSourceToTerraform = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const cdktf = require("cdktf");
function dataGithubRepositoryPagesSourceToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {};
}
exports.dataGithubRepositoryPagesSourceToTerraform = dataGithubRepositoryPagesSourceToTerraform;
class DataGithubRepositoryPagesSourceOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    * @param complexObjectIndex the index of this item in the list
    * @param complexObjectIsFromSet whether the list is wrapping a set (will add tolist() to be able to access an item via an index)
    */
    constructor(terraformResource, terraformAttribute, complexObjectIndex, complexObjectIsFromSet) {
        super(terraformResource, terraformAttribute, complexObjectIsFromSet, complexObjectIndex);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
        }
    }
    // branch - computed: true, optional: false, required: false
    get branch() {
        return this.getStringAttribute('branch');
    }
    // path - computed: true, optional: false, required: false
    get path() {
        return this.getStringAttribute('path');
    }
}
exports.DataGithubRepositoryPagesSourceOutputReference = DataGithubRepositoryPagesSourceOutputReference;
_a = JSII_RTTI_SYMBOL_1;
DataGithubRepositoryPagesSourceOutputReference[_a] = { fqn: "@cdktf/provider-github.dataGithubRepository.DataGithubRepositoryPagesSourceOutputReference", version: "12.0.0" };
class DataGithubRepositoryPagesSourceList extends cdktf.ComplexList {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    * @param wrapsSet whether the list is wrapping a set (will add tolist() to be able to access an item via an index)
    */
    constructor(terraformResource, terraformAttribute, wrapsSet) {
        super(terraformResource, terraformAttribute, wrapsSet);
        this.terraformResource = terraformResource;
        this.terraformAttribute = terraformAttribute;
        this.wrapsSet = wrapsSet;
    }
    /**
    * @param index the index of the item to return
    */
    get(index) {
        return new DataGithubRepositoryPagesSourceOutputReference(this.terraformResource, this.terraformAttribute, index, this.wrapsSet);
    }
}
exports.DataGithubRepositoryPagesSourceList = DataGithubRepositoryPagesSourceList;
_b = JSII_RTTI_SYMBOL_1;
DataGithubRepositoryPagesSourceList[_b] = { fqn: "@cdktf/provider-github.dataGithubRepository.DataGithubRepositoryPagesSourceList", version: "12.0.0" };
function dataGithubRepositoryPagesToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {};
}
exports.dataGithubRepositoryPagesToTerraform = dataGithubRepositoryPagesToTerraform;
class DataGithubRepositoryPagesOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    * @param complexObjectIndex the index of this item in the list
    * @param complexObjectIsFromSet whether the list is wrapping a set (will add tolist() to be able to access an item via an index)
    */
    constructor(terraformResource, terraformAttribute, complexObjectIndex, complexObjectIsFromSet) {
        super(terraformResource, terraformAttribute, complexObjectIsFromSet, complexObjectIndex);
        this.isEmptyObject = false;
        // source - computed: true, optional: false, required: false
        this._source = new DataGithubRepositoryPagesSourceList(this, "source", false);
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
        }
    }
    // build_type - computed: true, optional: false, required: false
    get buildType() {
        return this.getStringAttribute('build_type');
    }
    // cname - computed: true, optional: false, required: false
    get cname() {
        return this.getStringAttribute('cname');
    }
    // custom_404 - computed: true, optional: false, required: false
    get custom404() {
        return this.getBooleanAttribute('custom_404');
    }
    // html_url - computed: true, optional: false, required: false
    get htmlUrl() {
        return this.getStringAttribute('html_url');
    }
    get source() {
        return this._source;
    }
    // status - computed: true, optional: false, required: false
    get status() {
        return this.getStringAttribute('status');
    }
    // url - computed: true, optional: false, required: false
    get url() {
        return this.getStringAttribute('url');
    }
}
exports.DataGithubRepositoryPagesOutputReference = DataGithubRepositoryPagesOutputReference;
_c = JSII_RTTI_SYMBOL_1;
DataGithubRepositoryPagesOutputReference[_c] = { fqn: "@cdktf/provider-github.dataGithubRepository.DataGithubRepositoryPagesOutputReference", version: "12.0.0" };
class DataGithubRepositoryPagesList extends cdktf.ComplexList {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    * @param wrapsSet whether the list is wrapping a set (will add tolist() to be able to access an item via an index)
    */
    constructor(terraformResource, terraformAttribute, wrapsSet) {
        super(terraformResource, terraformAttribute, wrapsSet);
        this.terraformResource = terraformResource;
        this.terraformAttribute = terraformAttribute;
        this.wrapsSet = wrapsSet;
    }
    /**
    * @param index the index of the item to return
    */
    get(index) {
        return new DataGithubRepositoryPagesOutputReference(this.terraformResource, this.terraformAttribute, index, this.wrapsSet);
    }
}
exports.DataGithubRepositoryPagesList = DataGithubRepositoryPagesList;
_d = JSII_RTTI_SYMBOL_1;
DataGithubRepositoryPagesList[_d] = { fqn: "@cdktf/provider-github.dataGithubRepository.DataGithubRepositoryPagesList", version: "12.0.0" };
function dataGithubRepositoryTemplateToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {};
}
exports.dataGithubRepositoryTemplateToTerraform = dataGithubRepositoryTemplateToTerraform;
class DataGithubRepositoryTemplateOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    * @param complexObjectIndex the index of this item in the list
    * @param complexObjectIsFromSet whether the list is wrapping a set (will add tolist() to be able to access an item via an index)
    */
    constructor(terraformResource, terraformAttribute, complexObjectIndex, complexObjectIsFromSet) {
        super(terraformResource, terraformAttribute, complexObjectIsFromSet, complexObjectIndex);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
        }
    }
    // owner - computed: true, optional: false, required: false
    get owner() {
        return this.getStringAttribute('owner');
    }
    // repository - computed: true, optional: false, required: false
    get repository() {
        return this.getStringAttribute('repository');
    }
}
exports.DataGithubRepositoryTemplateOutputReference = DataGithubRepositoryTemplateOutputReference;
_e = JSII_RTTI_SYMBOL_1;
DataGithubRepositoryTemplateOutputReference[_e] = { fqn: "@cdktf/provider-github.dataGithubRepository.DataGithubRepositoryTemplateOutputReference", version: "12.0.0" };
class DataGithubRepositoryTemplateList extends cdktf.ComplexList {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    * @param wrapsSet whether the list is wrapping a set (will add tolist() to be able to access an item via an index)
    */
    constructor(terraformResource, terraformAttribute, wrapsSet) {
        super(terraformResource, terraformAttribute, wrapsSet);
        this.terraformResource = terraformResource;
        this.terraformAttribute = terraformAttribute;
        this.wrapsSet = wrapsSet;
    }
    /**
    * @param index the index of the item to return
    */
    get(index) {
        return new DataGithubRepositoryTemplateOutputReference(this.terraformResource, this.terraformAttribute, index, this.wrapsSet);
    }
}
exports.DataGithubRepositoryTemplateList = DataGithubRepositoryTemplateList;
_f = JSII_RTTI_SYMBOL_1;
DataGithubRepositoryTemplateList[_f] = { fqn: "@cdktf/provider-github.dataGithubRepository.DataGithubRepositoryTemplateList", version: "12.0.0" };
/**
* Represents a {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/repository github_repository}
*/
class DataGithubRepository extends cdktf.TerraformDataSource {
    // ==============
    // STATIC Methods
    // ==============
    /**
    * Generates CDKTF code for importing a DataGithubRepository resource upon running "cdktf plan <stack-name>"
    * @param scope The scope in which to define this construct
    * @param importToId The construct id used in the generated config for the DataGithubRepository to import
    * @param importFromId The id of the existing DataGithubRepository that should be imported. Refer to the {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/repository#import import section} in the documentation of this resource for the id to use
    * @param provider? Optional instance of the provider where the DataGithubRepository to import is found
    */
    static generateConfigForImport(scope, importToId, importFromId, provider) {
        return new cdktf.ImportableResource(scope, importToId, { terraformResourceType: "github_repository", importId: importFromId, provider });
    }
    // ===========
    // INITIALIZER
    // ===========
    /**
    * Create a new {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/repository github_repository} Data Source
    *
    * @param scope The scope in which to define this construct
    * @param id The scoped construct ID. Must be unique amongst siblings in the same scope
    * @param options DataGithubRepositoryConfig = {}
    */
    constructor(scope, id, config = {}) {
        super(scope, id, {
            terraformResourceType: 'github_repository',
            terraformGeneratorMetadata: {
                providerName: 'github',
                providerVersion: '5.40.0',
                providerVersionConstraint: '~> 5.0'
            },
            provider: config.provider,
            dependsOn: config.dependsOn,
            count: config.count,
            lifecycle: config.lifecycle,
            provisioners: config.provisioners,
            connection: config.connection,
            forEach: config.forEach
        });
        // pages - computed: true, optional: false, required: false
        this._pages = new DataGithubRepositoryPagesList(this, "pages", false);
        // template - computed: true, optional: false, required: false
        this._template = new DataGithubRepositoryTemplateList(this, "template", false);
        this._description = config.description;
        this._fullName = config.fullName;
        this._homepageUrl = config.homepageUrl;
        this._id = config.id;
        this._name = config.name;
    }
    // ==========
    // ATTRIBUTES
    // ==========
    // allow_auto_merge - computed: true, optional: false, required: false
    get allowAutoMerge() {
        return this.getBooleanAttribute('allow_auto_merge');
    }
    // allow_merge_commit - computed: true, optional: false, required: false
    get allowMergeCommit() {
        return this.getBooleanAttribute('allow_merge_commit');
    }
    // allow_rebase_merge - computed: true, optional: false, required: false
    get allowRebaseMerge() {
        return this.getBooleanAttribute('allow_rebase_merge');
    }
    // allow_squash_merge - computed: true, optional: false, required: false
    get allowSquashMerge() {
        return this.getBooleanAttribute('allow_squash_merge');
    }
    // archived - computed: true, optional: false, required: false
    get archived() {
        return this.getBooleanAttribute('archived');
    }
    // default_branch - computed: true, optional: false, required: false
    get defaultBranch() {
        return this.getStringAttribute('default_branch');
    }
    get description() {
        return this.getStringAttribute('description');
    }
    set description(value) {
        this._description = value;
    }
    resetDescription() {
        this._description = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get descriptionInput() {
        return this._description;
    }
    // fork - computed: true, optional: false, required: false
    get fork() {
        return this.getBooleanAttribute('fork');
    }
    get fullName() {
        return this.getStringAttribute('full_name');
    }
    set fullName(value) {
        this._fullName = value;
    }
    resetFullName() {
        this._fullName = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get fullNameInput() {
        return this._fullName;
    }
    // git_clone_url - computed: true, optional: false, required: false
    get gitCloneUrl() {
        return this.getStringAttribute('git_clone_url');
    }
    // has_discussions - computed: true, optional: false, required: false
    get hasDiscussions() {
        return this.getBooleanAttribute('has_discussions');
    }
    // has_downloads - computed: true, optional: false, required: false
    get hasDownloads() {
        return this.getBooleanAttribute('has_downloads');
    }
    // has_issues - computed: true, optional: false, required: false
    get hasIssues() {
        return this.getBooleanAttribute('has_issues');
    }
    // has_projects - computed: true, optional: false, required: false
    get hasProjects() {
        return this.getBooleanAttribute('has_projects');
    }
    // has_wiki - computed: true, optional: false, required: false
    get hasWiki() {
        return this.getBooleanAttribute('has_wiki');
    }
    get homepageUrl() {
        return this.getStringAttribute('homepage_url');
    }
    set homepageUrl(value) {
        this._homepageUrl = value;
    }
    resetHomepageUrl() {
        this._homepageUrl = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get homepageUrlInput() {
        return this._homepageUrl;
    }
    // html_url - computed: true, optional: false, required: false
    get htmlUrl() {
        return this.getStringAttribute('html_url');
    }
    // http_clone_url - computed: true, optional: false, required: false
    get httpCloneUrl() {
        return this.getStringAttribute('http_clone_url');
    }
    get id() {
        return this.getStringAttribute('id');
    }
    set id(value) {
        this._id = value;
    }
    resetId() {
        this._id = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get idInput() {
        return this._id;
    }
    // is_template - computed: true, optional: false, required: false
    get isTemplate() {
        return this.getBooleanAttribute('is_template');
    }
    // merge_commit_message - computed: true, optional: false, required: false
    get mergeCommitMessage() {
        return this.getStringAttribute('merge_commit_message');
    }
    // merge_commit_title - computed: true, optional: false, required: false
    get mergeCommitTitle() {
        return this.getStringAttribute('merge_commit_title');
    }
    get name() {
        return this.getStringAttribute('name');
    }
    set name(value) {
        this._name = value;
    }
    resetName() {
        this._name = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get nameInput() {
        return this._name;
    }
    // node_id - computed: true, optional: false, required: false
    get nodeId() {
        return this.getStringAttribute('node_id');
    }
    get pages() {
        return this._pages;
    }
    // primary_language - computed: true, optional: false, required: false
    get primaryLanguage() {
        return this.getStringAttribute('primary_language');
    }
    // private - computed: true, optional: false, required: false
    get private() {
        return this.getBooleanAttribute('private');
    }
    // repo_id - computed: true, optional: false, required: false
    get repoId() {
        return this.getNumberAttribute('repo_id');
    }
    // squash_merge_commit_message - computed: true, optional: false, required: false
    get squashMergeCommitMessage() {
        return this.getStringAttribute('squash_merge_commit_message');
    }
    // squash_merge_commit_title - computed: true, optional: false, required: false
    get squashMergeCommitTitle() {
        return this.getStringAttribute('squash_merge_commit_title');
    }
    // ssh_clone_url - computed: true, optional: false, required: false
    get sshCloneUrl() {
        return this.getStringAttribute('ssh_clone_url');
    }
    // svn_url - computed: true, optional: false, required: false
    get svnUrl() {
        return this.getStringAttribute('svn_url');
    }
    get template() {
        return this._template;
    }
    // topics - computed: true, optional: false, required: false
    get topics() {
        return this.getListAttribute('topics');
    }
    // visibility - computed: true, optional: false, required: false
    get visibility() {
        return this.getStringAttribute('visibility');
    }
    // =========
    // SYNTHESIS
    // =========
    synthesizeAttributes() {
        return {
            description: cdktf.stringToTerraform(this._description),
            full_name: cdktf.stringToTerraform(this._fullName),
            homepage_url: cdktf.stringToTerraform(this._homepageUrl),
            id: cdktf.stringToTerraform(this._id),
            name: cdktf.stringToTerraform(this._name),
        };
    }
}
exports.DataGithubRepository = DataGithubRepository;
_g = JSII_RTTI_SYMBOL_1;
DataGithubRepository[_g] = { fqn: "@cdktf/provider-github.dataGithubRepository.DataGithubRepository", version: "12.0.0" };
// =================
// STATIC PROPERTIES
// =================
DataGithubRepository.tfResourceType = "github_repository";
