// @cdktf/provider-github/lib/actions-organization-permissions/index.js
"use strict";
var _a, _b, _c;
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionsOrganizationPermissions = exports.ActionsOrganizationPermissionsEnabledRepositoriesConfigOutputReference = exports.actionsOrganizationPermissionsEnabledRepositoriesConfigToTerraform = exports.ActionsOrganizationPermissionsAllowedActionsConfigOutputReference = exports.actionsOrganizationPermissionsAllowedActionsConfigToTerraform = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const cdktf = require("cdktf");
function actionsOrganizationPermissionsAllowedActionsConfigToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        github_owned_allowed: cdktf.booleanToTerraform(struct.githubOwnedAllowed),
        patterns_allowed: cdktf.listMapper(cdktf.stringToTerraform, false)(struct.patternsAllowed),
        verified_allowed: cdktf.booleanToTerraform(struct.verifiedAllowed),
    };
}
exports.actionsOrganizationPermissionsAllowedActionsConfigToTerraform = actionsOrganizationPermissionsAllowedActionsConfigToTerraform;
class ActionsOrganizationPermissionsAllowedActionsConfigOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._githubOwnedAllowed !== undefined) {
            hasAnyValues = true;
            internalValueResult.githubOwnedAllowed = this._githubOwnedAllowed;
        }
        if (this._patternsAllowed !== undefined) {
            hasAnyValues = true;
            internalValueResult.patternsAllowed = this._patternsAllowed;
        }
        if (this._verifiedAllowed !== undefined) {
            hasAnyValues = true;
            internalValueResult.verifiedAllowed = this._verifiedAllowed;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._githubOwnedAllowed = undefined;
            this._patternsAllowed = undefined;
            this._verifiedAllowed = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._githubOwnedAllowed = value.githubOwnedAllowed;
            this._patternsAllowed = value.patternsAllowed;
            this._verifiedAllowed = value.verifiedAllowed;
        }
    }
    get githubOwnedAllowed() {
        return this.getBooleanAttribute('github_owned_allowed');
    }
    set githubOwnedAllowed(value) {
        this._githubOwnedAllowed = value;
    }
    // Temporarily expose input value. Use with caution.
    get githubOwnedAllowedInput() {
        return this._githubOwnedAllowed;
    }
    get patternsAllowed() {
        return cdktf.Fn.tolist(this.getListAttribute('patterns_allowed'));
    }
    set patternsAllowed(value) {
        this._patternsAllowed = value;
    }
    resetPatternsAllowed() {
        this._patternsAllowed = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get patternsAllowedInput() {
        return this._patternsAllowed;
    }
    get verifiedAllowed() {
        return this.getBooleanAttribute('verified_allowed');
    }
    set verifiedAllowed(value) {
        this._verifiedAllowed = value;
    }
    resetVerifiedAllowed() {
        this._verifiedAllowed = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get verifiedAllowedInput() {
        return this._verifiedAllowed;
    }
}
exports.ActionsOrganizationPermissionsAllowedActionsConfigOutputReference = ActionsOrganizationPermissionsAllowedActionsConfigOutputReference;
_a = JSII_RTTI_SYMBOL_1;
ActionsOrganizationPermissionsAllowedActionsConfigOutputReference[_a] = { fqn: "@cdktf/provider-github.actionsOrganizationPermissions.ActionsOrganizationPermissionsAllowedActionsConfigOutputReference", version: "12.0.0" };
function actionsOrganizationPermissionsEnabledRepositoriesConfigToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        repository_ids: cdktf.listMapper(cdktf.numberToTerraform, false)(struct.repositoryIds),
    };
}
exports.actionsOrganizationPermissionsEnabledRepositoriesConfigToTerraform = actionsOrganizationPermissionsEnabledRepositoriesConfigToTerraform;
class ActionsOrganizationPermissionsEnabledRepositoriesConfigOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._repositoryIds !== undefined) {
            hasAnyValues = true;
            internalValueResult.repositoryIds = this._repositoryIds;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._repositoryIds = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._repositoryIds = value.repositoryIds;
        }
    }
    get repositoryIds() {
        return cdktf.Token.asNumberList(cdktf.Fn.tolist(this.getNumberListAttribute('repository_ids')));
    }
    set repositoryIds(value) {
        this._repositoryIds = value;
    }
    // Temporarily expose input value. Use with caution.
    get repositoryIdsInput() {
        return this._repositoryIds;
    }
}
exports.ActionsOrganizationPermissionsEnabledRepositoriesConfigOutputReference = ActionsOrganizationPermissionsEnabledRepositoriesConfigOutputReference;
_b = JSII_RTTI_SYMBOL_1;
ActionsOrganizationPermissionsEnabledRepositoriesConfigOutputReference[_b] = { fqn: "@cdktf/provider-github.actionsOrganizationPermissions.ActionsOrganizationPermissionsEnabledRepositoriesConfigOutputReference", version: "12.0.0" };
/**
* Represents a {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/actions_organization_permissions github_actions_organization_permissions}
*/
class ActionsOrganizationPermissions extends cdktf.TerraformResource {
    // ==============
    // STATIC Methods
    // ==============
    /**
    * Generates CDKTF code for importing a ActionsOrganizationPermissions resource upon running "cdktf plan <stack-name>"
    * @param scope The scope in which to define this construct
    * @param importToId The construct id used in the generated config for the ActionsOrganizationPermissions to import
    * @param importFromId The id of the existing ActionsOrganizationPermissions that should be imported. Refer to the {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/actions_organization_permissions#import import section} in the documentation of this resource for the id to use
    * @param provider? Optional instance of the provider where the ActionsOrganizationPermissions to import is found
    */
    static generateConfigForImport(scope, importToId, importFromId, provider) {
        return new cdktf.ImportableResource(scope, importToId, { terraformResourceType: "github_actions_organization_permissions", importId: importFromId, provider });
    }
    // ===========
    // INITIALIZER
    // ===========
    /**
    * Create a new {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/actions_organization_permissions github_actions_organization_permissions} Resource
    *
    * @param scope The scope in which to define this construct
    * @param id The scoped construct ID. Must be unique amongst siblings in the same scope
    * @param options ActionsOrganizationPermissionsConfig
    */
    constructor(scope, id, config) {
        super(scope, id, {
            terraformResourceType: 'github_actions_organization_permissions',
            terraformGeneratorMetadata: {
                providerName: 'github',
                providerVersion: '5.40.0',
                providerVersionConstraint: '~> 5.0'
            },
            provider: config.provider,
            dependsOn: config.dependsOn,
            count: config.count,
            lifecycle: config.lifecycle,
            provisioners: config.provisioners,
            connection: config.connection,
            forEach: config.forEach
        });
        // allowed_actions_config - computed: false, optional: true, required: false
        this._allowedActionsConfig = new ActionsOrganizationPermissionsAllowedActionsConfigOutputReference(this, "allowed_actions_config");
        // enabled_repositories_config - computed: false, optional: true, required: false
        this._enabledRepositoriesConfig = new ActionsOrganizationPermissionsEnabledRepositoriesConfigOutputReference(this, "enabled_repositories_config");
        this._allowedActions = config.allowedActions;
        this._enabledRepositories = config.enabledRepositories;
        this._id = config.id;
        this._allowedActionsConfig.internalValue = config.allowedActionsConfig;
        this._enabledRepositoriesConfig.internalValue = config.enabledRepositoriesConfig;
    }
    get allowedActions() {
        return this.getStringAttribute('allowed_actions');
    }
    set allowedActions(value) {
        this._allowedActions = value;
    }
    resetAllowedActions() {
        this._allowedActions = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get allowedActionsInput() {
        return this._allowedActions;
    }
    get enabledRepositories() {
        return this.getStringAttribute('enabled_repositories');
    }
    set enabledRepositories(value) {
        this._enabledRepositories = value;
    }
    // Temporarily expose input value. Use with caution.
    get enabledRepositoriesInput() {
        return this._enabledRepositories;
    }
    get id() {
        return this.getStringAttribute('id');
    }
    set id(value) {
        this._id = value;
    }
    resetId() {
        this._id = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get idInput() {
        return this._id;
    }
    get allowedActionsConfig() {
        return this._allowedActionsConfig;
    }
    putAllowedActionsConfig(value) {
        this._allowedActionsConfig.internalValue = value;
    }
    resetAllowedActionsConfig() {
        this._allowedActionsConfig.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get allowedActionsConfigInput() {
        return this._allowedActionsConfig.internalValue;
    }
    get enabledRepositoriesConfig() {
        return this._enabledRepositoriesConfig;
    }
    putEnabledRepositoriesConfig(value) {
        this._enabledRepositoriesConfig.internalValue = value;
    }
    resetEnabledRepositoriesConfig() {
        this._enabledRepositoriesConfig.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get enabledRepositoriesConfigInput() {
        return this._enabledRepositoriesConfig.internalValue;
    }
    // =========
    // SYNTHESIS
    // =========
    synthesizeAttributes() {
        return {
            allowed_actions: cdktf.stringToTerraform(this._allowedActions),
            enabled_repositories: cdktf.stringToTerraform(this._enabledRepositories),
            id: cdktf.stringToTerraform(this._id),
            allowed_actions_config: actionsOrganizationPermissionsAllowedActionsConfigToTerraform(this._allowedActionsConfig.internalValue),
            enabled_repositories_config: actionsOrganizationPermissionsEnabledRepositoriesConfigToTerraform(this._enabledRepositoriesConfig.internalValue),
        };
    }
}
exports.ActionsOrganizationPermissions = ActionsOrganizationPermissions;
_c = JSII_RTTI_SYMBOL_1;
ActionsOrganizationPermissions[_c] = { fqn: "@cdktf/provider-github.actionsOrganizationPermissions.ActionsOrganizationPermissions", version: "12.0.0" };
// =================
// STATIC PROPERTIES
// =================
ActionsOrganizationPermissions.tfResourceType = "github_actions_organization_permissions";
