// @cdktf/provider-github/lib/data-github-organization-external-identities/index.js
"use strict";
var _a, _b, _c;
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataGithubOrganizationExternalIdentities = exports.DataGithubOrganizationExternalIdentitiesIdentitiesList = exports.DataGithubOrganizationExternalIdentitiesIdentitiesOutputReference = exports.dataGithubOrganizationExternalIdentitiesIdentitiesToTerraform = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const cdktf = require("cdktf");
function dataGithubOrganizationExternalIdentitiesIdentitiesToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {};
}
exports.dataGithubOrganizationExternalIdentitiesIdentitiesToTerraform = dataGithubOrganizationExternalIdentitiesIdentitiesToTerraform;
class DataGithubOrganizationExternalIdentitiesIdentitiesOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    * @param complexObjectIndex the index of this item in the list
    * @param complexObjectIsFromSet whether the list is wrapping a set (will add tolist() to be able to access an item via an index)
    */
    constructor(terraformResource, terraformAttribute, complexObjectIndex, complexObjectIsFromSet) {
        super(terraformResource, terraformAttribute, complexObjectIsFromSet, complexObjectIndex);
        this.isEmptyObject = false;
        // saml_identity - computed: true, optional: false, required: false
        this._samlIdentity = new cdktf.StringMap(this, "saml_identity");
        // scim_identity - computed: true, optional: false, required: false
        this._scimIdentity = new cdktf.StringMap(this, "scim_identity");
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
        }
    }
    // login - computed: true, optional: false, required: false
    get login() {
        return this.getStringAttribute('login');
    }
    get samlIdentity() {
        return this._samlIdentity;
    }
    get scimIdentity() {
        return this._scimIdentity;
    }
}
exports.DataGithubOrganizationExternalIdentitiesIdentitiesOutputReference = DataGithubOrganizationExternalIdentitiesIdentitiesOutputReference;
_a = JSII_RTTI_SYMBOL_1;
DataGithubOrganizationExternalIdentitiesIdentitiesOutputReference[_a] = { fqn: "@cdktf/provider-github.dataGithubOrganizationExternalIdentities.DataGithubOrganizationExternalIdentitiesIdentitiesOutputReference", version: "12.0.0" };
class DataGithubOrganizationExternalIdentitiesIdentitiesList extends cdktf.ComplexList {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    * @param wrapsSet whether the list is wrapping a set (will add tolist() to be able to access an item via an index)
    */
    constructor(terraformResource, terraformAttribute, wrapsSet) {
        super(terraformResource, terraformAttribute, wrapsSet);
        this.terraformResource = terraformResource;
        this.terraformAttribute = terraformAttribute;
        this.wrapsSet = wrapsSet;
    }
    /**
    * @param index the index of the item to return
    */
    get(index) {
        return new DataGithubOrganizationExternalIdentitiesIdentitiesOutputReference(this.terraformResource, this.terraformAttribute, index, this.wrapsSet);
    }
}
exports.DataGithubOrganizationExternalIdentitiesIdentitiesList = DataGithubOrganizationExternalIdentitiesIdentitiesList;
_b = JSII_RTTI_SYMBOL_1;
DataGithubOrganizationExternalIdentitiesIdentitiesList[_b] = { fqn: "@cdktf/provider-github.dataGithubOrganizationExternalIdentities.DataGithubOrganizationExternalIdentitiesIdentitiesList", version: "12.0.0" };
/**
* Represents a {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/organization_external_identities github_organization_external_identities}
*/
class DataGithubOrganizationExternalIdentities extends cdktf.TerraformDataSource {
    // ==============
    // STATIC Methods
    // ==============
    /**
    * Generates CDKTF code for importing a DataGithubOrganizationExternalIdentities resource upon running "cdktf plan <stack-name>"
    * @param scope The scope in which to define this construct
    * @param importToId The construct id used in the generated config for the DataGithubOrganizationExternalIdentities to import
    * @param importFromId The id of the existing DataGithubOrganizationExternalIdentities that should be imported. Refer to the {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/organization_external_identities#import import section} in the documentation of this resource for the id to use
    * @param provider? Optional instance of the provider where the DataGithubOrganizationExternalIdentities to import is found
    */
    static generateConfigForImport(scope, importToId, importFromId, provider) {
        return new cdktf.ImportableResource(scope, importToId, { terraformResourceType: "github_organization_external_identities", importId: importFromId, provider });
    }
    // ===========
    // INITIALIZER
    // ===========
    /**
    * Create a new {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/data-sources/organization_external_identities github_organization_external_identities} Data Source
    *
    * @param scope The scope in which to define this construct
    * @param id The scoped construct ID. Must be unique amongst siblings in the same scope
    * @param options DataGithubOrganizationExternalIdentitiesConfig = {}
    */
    constructor(scope, id, config = {}) {
        super(scope, id, {
            terraformResourceType: 'github_organization_external_identities',
            terraformGeneratorMetadata: {
                providerName: 'github',
                providerVersion: '5.40.0',
                providerVersionConstraint: '~> 5.0'
            },
            provider: config.provider,
            dependsOn: config.dependsOn,
            count: config.count,
            lifecycle: config.lifecycle,
            provisioners: config.provisioners,
            connection: config.connection,
            forEach: config.forEach
        });
        // identities - computed: true, optional: false, required: false
        this._identities = new DataGithubOrganizationExternalIdentitiesIdentitiesList(this, "identities", false);
        this._id = config.id;
    }
    get id() {
        return this.getStringAttribute('id');
    }
    set id(value) {
        this._id = value;
    }
    resetId() {
        this._id = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get idInput() {
        return this._id;
    }
    get identities() {
        return this._identities;
    }
    // =========
    // SYNTHESIS
    // =========
    synthesizeAttributes() {
        return {
            id: cdktf.stringToTerraform(this._id),
        };
    }
}
exports.DataGithubOrganizationExternalIdentities = DataGithubOrganizationExternalIdentities;
_c = JSII_RTTI_SYMBOL_1;
DataGithubOrganizationExternalIdentities[_c] = { fqn: "@cdktf/provider-github.dataGithubOrganizationExternalIdentities.DataGithubOrganizationExternalIdentities", version: "12.0.0" };
// =================
// STATIC PROPERTIES
// =================
DataGithubOrganizationExternalIdentities.tfResourceType = "github_organization_external_identities";
