// @cdktf/provider-github/lib/repository/index.js
"use strict";
var _a, _b, _c, _d, _e, _f, _g, _h;
Object.defineProperty(exports, "__esModule", { value: true });
exports.Repository = exports.RepositoryTemplateOutputReference = exports.repositoryTemplateToTerraform = exports.RepositorySecurityAndAnalysisOutputReference = exports.repositorySecurityAndAnalysisToTerraform = exports.RepositorySecurityAndAnalysisSecretScanningPushProtectionOutputReference = exports.repositorySecurityAndAnalysisSecretScanningPushProtectionToTerraform = exports.RepositorySecurityAndAnalysisSecretScanningOutputReference = exports.repositorySecurityAndAnalysisSecretScanningToTerraform = exports.RepositorySecurityAndAnalysisAdvancedSecurityOutputReference = exports.repositorySecurityAndAnalysisAdvancedSecurityToTerraform = exports.RepositoryPagesOutputReference = exports.repositoryPagesToTerraform = exports.RepositoryPagesSourceOutputReference = exports.repositoryPagesSourceToTerraform = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const cdktf = require("cdktf");
function repositoryPagesSourceToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        branch: cdktf.stringToTerraform(struct.branch),
        path: cdktf.stringToTerraform(struct.path),
    };
}
exports.repositoryPagesSourceToTerraform = repositoryPagesSourceToTerraform;
class RepositoryPagesSourceOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._branch !== undefined) {
            hasAnyValues = true;
            internalValueResult.branch = this._branch;
        }
        if (this._path !== undefined) {
            hasAnyValues = true;
            internalValueResult.path = this._path;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._branch = undefined;
            this._path = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._branch = value.branch;
            this._path = value.path;
        }
    }
    get branch() {
        return this.getStringAttribute('branch');
    }
    set branch(value) {
        this._branch = value;
    }
    // Temporarily expose input value. Use with caution.
    get branchInput() {
        return this._branch;
    }
    get path() {
        return this.getStringAttribute('path');
    }
    set path(value) {
        this._path = value;
    }
    resetPath() {
        this._path = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get pathInput() {
        return this._path;
    }
}
exports.RepositoryPagesSourceOutputReference = RepositoryPagesSourceOutputReference;
_a = JSII_RTTI_SYMBOL_1;
RepositoryPagesSourceOutputReference[_a] = { fqn: "@cdktf/provider-github.repository.RepositoryPagesSourceOutputReference", version: "12.0.0" };
function repositoryPagesToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        build_type: cdktf.stringToTerraform(struct.buildType),
        cname: cdktf.stringToTerraform(struct.cname),
        source: repositoryPagesSourceToTerraform(struct.source),
    };
}
exports.repositoryPagesToTerraform = repositoryPagesToTerraform;
class RepositoryPagesOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
        // source - computed: false, optional: true, required: false
        this._source = new RepositoryPagesSourceOutputReference(this, "source");
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._buildType !== undefined) {
            hasAnyValues = true;
            internalValueResult.buildType = this._buildType;
        }
        if (this._cname !== undefined) {
            hasAnyValues = true;
            internalValueResult.cname = this._cname;
        }
        if (this._source?.internalValue !== undefined) {
            hasAnyValues = true;
            internalValueResult.source = this._source?.internalValue;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._buildType = undefined;
            this._cname = undefined;
            this._source.internalValue = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._buildType = value.buildType;
            this._cname = value.cname;
            this._source.internalValue = value.source;
        }
    }
    get buildType() {
        return this.getStringAttribute('build_type');
    }
    set buildType(value) {
        this._buildType = value;
    }
    resetBuildType() {
        this._buildType = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get buildTypeInput() {
        return this._buildType;
    }
    get cname() {
        return this.getStringAttribute('cname');
    }
    set cname(value) {
        this._cname = value;
    }
    resetCname() {
        this._cname = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get cnameInput() {
        return this._cname;
    }
    // custom_404 - computed: true, optional: false, required: false
    get custom404() {
        return this.getBooleanAttribute('custom_404');
    }
    // html_url - computed: true, optional: false, required: false
    get htmlUrl() {
        return this.getStringAttribute('html_url');
    }
    // status - computed: true, optional: false, required: false
    get status() {
        return this.getStringAttribute('status');
    }
    // url - computed: true, optional: false, required: false
    get url() {
        return this.getStringAttribute('url');
    }
    get source() {
        return this._source;
    }
    putSource(value) {
        this._source.internalValue = value;
    }
    resetSource() {
        this._source.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get sourceInput() {
        return this._source.internalValue;
    }
}
exports.RepositoryPagesOutputReference = RepositoryPagesOutputReference;
_b = JSII_RTTI_SYMBOL_1;
RepositoryPagesOutputReference[_b] = { fqn: "@cdktf/provider-github.repository.RepositoryPagesOutputReference", version: "12.0.0" };
function repositorySecurityAndAnalysisAdvancedSecurityToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        status: cdktf.stringToTerraform(struct.status),
    };
}
exports.repositorySecurityAndAnalysisAdvancedSecurityToTerraform = repositorySecurityAndAnalysisAdvancedSecurityToTerraform;
class RepositorySecurityAndAnalysisAdvancedSecurityOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._status !== undefined) {
            hasAnyValues = true;
            internalValueResult.status = this._status;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._status = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._status = value.status;
        }
    }
    get status() {
        return this.getStringAttribute('status');
    }
    set status(value) {
        this._status = value;
    }
    // Temporarily expose input value. Use with caution.
    get statusInput() {
        return this._status;
    }
}
exports.RepositorySecurityAndAnalysisAdvancedSecurityOutputReference = RepositorySecurityAndAnalysisAdvancedSecurityOutputReference;
_c = JSII_RTTI_SYMBOL_1;
RepositorySecurityAndAnalysisAdvancedSecurityOutputReference[_c] = { fqn: "@cdktf/provider-github.repository.RepositorySecurityAndAnalysisAdvancedSecurityOutputReference", version: "12.0.0" };
function repositorySecurityAndAnalysisSecretScanningToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        status: cdktf.stringToTerraform(struct.status),
    };
}
exports.repositorySecurityAndAnalysisSecretScanningToTerraform = repositorySecurityAndAnalysisSecretScanningToTerraform;
class RepositorySecurityAndAnalysisSecretScanningOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._status !== undefined) {
            hasAnyValues = true;
            internalValueResult.status = this._status;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._status = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._status = value.status;
        }
    }
    get status() {
        return this.getStringAttribute('status');
    }
    set status(value) {
        this._status = value;
    }
    // Temporarily expose input value. Use with caution.
    get statusInput() {
        return this._status;
    }
}
exports.RepositorySecurityAndAnalysisSecretScanningOutputReference = RepositorySecurityAndAnalysisSecretScanningOutputReference;
_d = JSII_RTTI_SYMBOL_1;
RepositorySecurityAndAnalysisSecretScanningOutputReference[_d] = { fqn: "@cdktf/provider-github.repository.RepositorySecurityAndAnalysisSecretScanningOutputReference", version: "12.0.0" };
function repositorySecurityAndAnalysisSecretScanningPushProtectionToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        status: cdktf.stringToTerraform(struct.status),
    };
}
exports.repositorySecurityAndAnalysisSecretScanningPushProtectionToTerraform = repositorySecurityAndAnalysisSecretScanningPushProtectionToTerraform;
class RepositorySecurityAndAnalysisSecretScanningPushProtectionOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._status !== undefined) {
            hasAnyValues = true;
            internalValueResult.status = this._status;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._status = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._status = value.status;
        }
    }
    get status() {
        return this.getStringAttribute('status');
    }
    set status(value) {
        this._status = value;
    }
    // Temporarily expose input value. Use with caution.
    get statusInput() {
        return this._status;
    }
}
exports.RepositorySecurityAndAnalysisSecretScanningPushProtectionOutputReference = RepositorySecurityAndAnalysisSecretScanningPushProtectionOutputReference;
_e = JSII_RTTI_SYMBOL_1;
RepositorySecurityAndAnalysisSecretScanningPushProtectionOutputReference[_e] = { fqn: "@cdktf/provider-github.repository.RepositorySecurityAndAnalysisSecretScanningPushProtectionOutputReference", version: "12.0.0" };
function repositorySecurityAndAnalysisToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        advanced_security: repositorySecurityAndAnalysisAdvancedSecurityToTerraform(struct.advancedSecurity),
        secret_scanning: repositorySecurityAndAnalysisSecretScanningToTerraform(struct.secretScanning),
        secret_scanning_push_protection: repositorySecurityAndAnalysisSecretScanningPushProtectionToTerraform(struct.secretScanningPushProtection),
    };
}
exports.repositorySecurityAndAnalysisToTerraform = repositorySecurityAndAnalysisToTerraform;
class RepositorySecurityAndAnalysisOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
        // advanced_security - computed: false, optional: true, required: false
        this._advancedSecurity = new RepositorySecurityAndAnalysisAdvancedSecurityOutputReference(this, "advanced_security");
        // secret_scanning - computed: false, optional: true, required: false
        this._secretScanning = new RepositorySecurityAndAnalysisSecretScanningOutputReference(this, "secret_scanning");
        // secret_scanning_push_protection - computed: false, optional: true, required: false
        this._secretScanningPushProtection = new RepositorySecurityAndAnalysisSecretScanningPushProtectionOutputReference(this, "secret_scanning_push_protection");
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._advancedSecurity?.internalValue !== undefined) {
            hasAnyValues = true;
            internalValueResult.advancedSecurity = this._advancedSecurity?.internalValue;
        }
        if (this._secretScanning?.internalValue !== undefined) {
            hasAnyValues = true;
            internalValueResult.secretScanning = this._secretScanning?.internalValue;
        }
        if (this._secretScanningPushProtection?.internalValue !== undefined) {
            hasAnyValues = true;
            internalValueResult.secretScanningPushProtection = this._secretScanningPushProtection?.internalValue;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._advancedSecurity.internalValue = undefined;
            this._secretScanning.internalValue = undefined;
            this._secretScanningPushProtection.internalValue = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._advancedSecurity.internalValue = value.advancedSecurity;
            this._secretScanning.internalValue = value.secretScanning;
            this._secretScanningPushProtection.internalValue = value.secretScanningPushProtection;
        }
    }
    get advancedSecurity() {
        return this._advancedSecurity;
    }
    putAdvancedSecurity(value) {
        this._advancedSecurity.internalValue = value;
    }
    resetAdvancedSecurity() {
        this._advancedSecurity.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get advancedSecurityInput() {
        return this._advancedSecurity.internalValue;
    }
    get secretScanning() {
        return this._secretScanning;
    }
    putSecretScanning(value) {
        this._secretScanning.internalValue = value;
    }
    resetSecretScanning() {
        this._secretScanning.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get secretScanningInput() {
        return this._secretScanning.internalValue;
    }
    get secretScanningPushProtection() {
        return this._secretScanningPushProtection;
    }
    putSecretScanningPushProtection(value) {
        this._secretScanningPushProtection.internalValue = value;
    }
    resetSecretScanningPushProtection() {
        this._secretScanningPushProtection.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get secretScanningPushProtectionInput() {
        return this._secretScanningPushProtection.internalValue;
    }
}
exports.RepositorySecurityAndAnalysisOutputReference = RepositorySecurityAndAnalysisOutputReference;
_f = JSII_RTTI_SYMBOL_1;
RepositorySecurityAndAnalysisOutputReference[_f] = { fqn: "@cdktf/provider-github.repository.RepositorySecurityAndAnalysisOutputReference", version: "12.0.0" };
function repositoryTemplateToTerraform(struct) {
    if (!cdktf.canInspect(struct) || cdktf.Tokenization.isResolvable(struct)) {
        return struct;
    }
    if (cdktf.isComplexElement(struct)) {
        throw new Error("A complex element was used as configuration, this is not supported: https://cdk.tf/complex-object-as-configuration");
    }
    return {
        include_all_branches: cdktf.booleanToTerraform(struct.includeAllBranches),
        owner: cdktf.stringToTerraform(struct.owner),
        repository: cdktf.stringToTerraform(struct.repository),
    };
}
exports.repositoryTemplateToTerraform = repositoryTemplateToTerraform;
class RepositoryTemplateOutputReference extends cdktf.ComplexObject {
    /**
    * @param terraformResource The parent resource
    * @param terraformAttribute The attribute on the parent resource this class is referencing
    */
    constructor(terraformResource, terraformAttribute) {
        super(terraformResource, terraformAttribute, false, 0);
        this.isEmptyObject = false;
    }
    get internalValue() {
        let hasAnyValues = this.isEmptyObject;
        const internalValueResult = {};
        if (this._includeAllBranches !== undefined) {
            hasAnyValues = true;
            internalValueResult.includeAllBranches = this._includeAllBranches;
        }
        if (this._owner !== undefined) {
            hasAnyValues = true;
            internalValueResult.owner = this._owner;
        }
        if (this._repository !== undefined) {
            hasAnyValues = true;
            internalValueResult.repository = this._repository;
        }
        return hasAnyValues ? internalValueResult : undefined;
    }
    set internalValue(value) {
        if (value === undefined) {
            this.isEmptyObject = false;
            this._includeAllBranches = undefined;
            this._owner = undefined;
            this._repository = undefined;
        }
        else {
            this.isEmptyObject = Object.keys(value).length === 0;
            this._includeAllBranches = value.includeAllBranches;
            this._owner = value.owner;
            this._repository = value.repository;
        }
    }
    get includeAllBranches() {
        return this.getBooleanAttribute('include_all_branches');
    }
    set includeAllBranches(value) {
        this._includeAllBranches = value;
    }
    resetIncludeAllBranches() {
        this._includeAllBranches = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get includeAllBranchesInput() {
        return this._includeAllBranches;
    }
    get owner() {
        return this.getStringAttribute('owner');
    }
    set owner(value) {
        this._owner = value;
    }
    // Temporarily expose input value. Use with caution.
    get ownerInput() {
        return this._owner;
    }
    get repository() {
        return this.getStringAttribute('repository');
    }
    set repository(value) {
        this._repository = value;
    }
    // Temporarily expose input value. Use with caution.
    get repositoryInput() {
        return this._repository;
    }
}
exports.RepositoryTemplateOutputReference = RepositoryTemplateOutputReference;
_g = JSII_RTTI_SYMBOL_1;
RepositoryTemplateOutputReference[_g] = { fqn: "@cdktf/provider-github.repository.RepositoryTemplateOutputReference", version: "12.0.0" };
/**
* Represents a {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/repository github_repository}
*/
class Repository extends cdktf.TerraformResource {
    // ==============
    // STATIC Methods
    // ==============
    /**
    * Generates CDKTF code for importing a Repository resource upon running "cdktf plan <stack-name>"
    * @param scope The scope in which to define this construct
    * @param importToId The construct id used in the generated config for the Repository to import
    * @param importFromId The id of the existing Repository that should be imported. Refer to the {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/repository#import import section} in the documentation of this resource for the id to use
    * @param provider? Optional instance of the provider where the Repository to import is found
    */
    static generateConfigForImport(scope, importToId, importFromId, provider) {
        return new cdktf.ImportableResource(scope, importToId, { terraformResourceType: "github_repository", importId: importFromId, provider });
    }
    // ===========
    // INITIALIZER
    // ===========
    /**
    * Create a new {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/repository github_repository} Resource
    *
    * @param scope The scope in which to define this construct
    * @param id The scoped construct ID. Must be unique amongst siblings in the same scope
    * @param options RepositoryConfig
    */
    constructor(scope, id, config) {
        super(scope, id, {
            terraformResourceType: 'github_repository',
            terraformGeneratorMetadata: {
                providerName: 'github',
                providerVersion: '5.40.0',
                providerVersionConstraint: '~> 5.0'
            },
            provider: config.provider,
            dependsOn: config.dependsOn,
            count: config.count,
            lifecycle: config.lifecycle,
            provisioners: config.provisioners,
            connection: config.connection,
            forEach: config.forEach
        });
        // pages - computed: false, optional: true, required: false
        this._pages = new RepositoryPagesOutputReference(this, "pages");
        // security_and_analysis - computed: false, optional: true, required: false
        this._securityAndAnalysis = new RepositorySecurityAndAnalysisOutputReference(this, "security_and_analysis");
        // template - computed: false, optional: true, required: false
        this._template = new RepositoryTemplateOutputReference(this, "template");
        this._allowAutoMerge = config.allowAutoMerge;
        this._allowMergeCommit = config.allowMergeCommit;
        this._allowRebaseMerge = config.allowRebaseMerge;
        this._allowSquashMerge = config.allowSquashMerge;
        this._allowUpdateBranch = config.allowUpdateBranch;
        this._archiveOnDestroy = config.archiveOnDestroy;
        this._archived = config.archived;
        this._autoInit = config.autoInit;
        this._defaultBranch = config.defaultBranch;
        this._deleteBranchOnMerge = config.deleteBranchOnMerge;
        this._description = config.description;
        this._gitignoreTemplate = config.gitignoreTemplate;
        this._hasDiscussions = config.hasDiscussions;
        this._hasDownloads = config.hasDownloads;
        this._hasIssues = config.hasIssues;
        this._hasProjects = config.hasProjects;
        this._hasWiki = config.hasWiki;
        this._homepageUrl = config.homepageUrl;
        this._id = config.id;
        this._ignoreVulnerabilityAlertsDuringRead = config.ignoreVulnerabilityAlertsDuringRead;
        this._isTemplate = config.isTemplate;
        this._licenseTemplate = config.licenseTemplate;
        this._mergeCommitMessage = config.mergeCommitMessage;
        this._mergeCommitTitle = config.mergeCommitTitle;
        this._name = config.name;
        this._private = config.private;
        this._squashMergeCommitMessage = config.squashMergeCommitMessage;
        this._squashMergeCommitTitle = config.squashMergeCommitTitle;
        this._topics = config.topics;
        this._visibility = config.visibility;
        this._vulnerabilityAlerts = config.vulnerabilityAlerts;
        this._pages.internalValue = config.pages;
        this._securityAndAnalysis.internalValue = config.securityAndAnalysis;
        this._template.internalValue = config.template;
    }
    get allowAutoMerge() {
        return this.getBooleanAttribute('allow_auto_merge');
    }
    set allowAutoMerge(value) {
        this._allowAutoMerge = value;
    }
    resetAllowAutoMerge() {
        this._allowAutoMerge = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get allowAutoMergeInput() {
        return this._allowAutoMerge;
    }
    get allowMergeCommit() {
        return this.getBooleanAttribute('allow_merge_commit');
    }
    set allowMergeCommit(value) {
        this._allowMergeCommit = value;
    }
    resetAllowMergeCommit() {
        this._allowMergeCommit = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get allowMergeCommitInput() {
        return this._allowMergeCommit;
    }
    get allowRebaseMerge() {
        return this.getBooleanAttribute('allow_rebase_merge');
    }
    set allowRebaseMerge(value) {
        this._allowRebaseMerge = value;
    }
    resetAllowRebaseMerge() {
        this._allowRebaseMerge = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get allowRebaseMergeInput() {
        return this._allowRebaseMerge;
    }
    get allowSquashMerge() {
        return this.getBooleanAttribute('allow_squash_merge');
    }
    set allowSquashMerge(value) {
        this._allowSquashMerge = value;
    }
    resetAllowSquashMerge() {
        this._allowSquashMerge = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get allowSquashMergeInput() {
        return this._allowSquashMerge;
    }
    get allowUpdateBranch() {
        return this.getBooleanAttribute('allow_update_branch');
    }
    set allowUpdateBranch(value) {
        this._allowUpdateBranch = value;
    }
    resetAllowUpdateBranch() {
        this._allowUpdateBranch = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get allowUpdateBranchInput() {
        return this._allowUpdateBranch;
    }
    get archiveOnDestroy() {
        return this.getBooleanAttribute('archive_on_destroy');
    }
    set archiveOnDestroy(value) {
        this._archiveOnDestroy = value;
    }
    resetArchiveOnDestroy() {
        this._archiveOnDestroy = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get archiveOnDestroyInput() {
        return this._archiveOnDestroy;
    }
    get archived() {
        return this.getBooleanAttribute('archived');
    }
    set archived(value) {
        this._archived = value;
    }
    resetArchived() {
        this._archived = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get archivedInput() {
        return this._archived;
    }
    get autoInit() {
        return this.getBooleanAttribute('auto_init');
    }
    set autoInit(value) {
        this._autoInit = value;
    }
    resetAutoInit() {
        this._autoInit = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get autoInitInput() {
        return this._autoInit;
    }
    get defaultBranch() {
        return this.getStringAttribute('default_branch');
    }
    set defaultBranch(value) {
        this._defaultBranch = value;
    }
    resetDefaultBranch() {
        this._defaultBranch = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get defaultBranchInput() {
        return this._defaultBranch;
    }
    get deleteBranchOnMerge() {
        return this.getBooleanAttribute('delete_branch_on_merge');
    }
    set deleteBranchOnMerge(value) {
        this._deleteBranchOnMerge = value;
    }
    resetDeleteBranchOnMerge() {
        this._deleteBranchOnMerge = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get deleteBranchOnMergeInput() {
        return this._deleteBranchOnMerge;
    }
    get description() {
        return this.getStringAttribute('description');
    }
    set description(value) {
        this._description = value;
    }
    resetDescription() {
        this._description = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get descriptionInput() {
        return this._description;
    }
    // etag - computed: true, optional: false, required: false
    get etag() {
        return this.getStringAttribute('etag');
    }
    // full_name - computed: true, optional: false, required: false
    get fullName() {
        return this.getStringAttribute('full_name');
    }
    // git_clone_url - computed: true, optional: false, required: false
    get gitCloneUrl() {
        return this.getStringAttribute('git_clone_url');
    }
    get gitignoreTemplate() {
        return this.getStringAttribute('gitignore_template');
    }
    set gitignoreTemplate(value) {
        this._gitignoreTemplate = value;
    }
    resetGitignoreTemplate() {
        this._gitignoreTemplate = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get gitignoreTemplateInput() {
        return this._gitignoreTemplate;
    }
    get hasDiscussions() {
        return this.getBooleanAttribute('has_discussions');
    }
    set hasDiscussions(value) {
        this._hasDiscussions = value;
    }
    resetHasDiscussions() {
        this._hasDiscussions = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get hasDiscussionsInput() {
        return this._hasDiscussions;
    }
    get hasDownloads() {
        return this.getBooleanAttribute('has_downloads');
    }
    set hasDownloads(value) {
        this._hasDownloads = value;
    }
    resetHasDownloads() {
        this._hasDownloads = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get hasDownloadsInput() {
        return this._hasDownloads;
    }
    get hasIssues() {
        return this.getBooleanAttribute('has_issues');
    }
    set hasIssues(value) {
        this._hasIssues = value;
    }
    resetHasIssues() {
        this._hasIssues = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get hasIssuesInput() {
        return this._hasIssues;
    }
    get hasProjects() {
        return this.getBooleanAttribute('has_projects');
    }
    set hasProjects(value) {
        this._hasProjects = value;
    }
    resetHasProjects() {
        this._hasProjects = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get hasProjectsInput() {
        return this._hasProjects;
    }
    get hasWiki() {
        return this.getBooleanAttribute('has_wiki');
    }
    set hasWiki(value) {
        this._hasWiki = value;
    }
    resetHasWiki() {
        this._hasWiki = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get hasWikiInput() {
        return this._hasWiki;
    }
    get homepageUrl() {
        return this.getStringAttribute('homepage_url');
    }
    set homepageUrl(value) {
        this._homepageUrl = value;
    }
    resetHomepageUrl() {
        this._homepageUrl = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get homepageUrlInput() {
        return this._homepageUrl;
    }
    // html_url - computed: true, optional: false, required: false
    get htmlUrl() {
        return this.getStringAttribute('html_url');
    }
    // http_clone_url - computed: true, optional: false, required: false
    get httpCloneUrl() {
        return this.getStringAttribute('http_clone_url');
    }
    get id() {
        return this.getStringAttribute('id');
    }
    set id(value) {
        this._id = value;
    }
    resetId() {
        this._id = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get idInput() {
        return this._id;
    }
    get ignoreVulnerabilityAlertsDuringRead() {
        return this.getBooleanAttribute('ignore_vulnerability_alerts_during_read');
    }
    set ignoreVulnerabilityAlertsDuringRead(value) {
        this._ignoreVulnerabilityAlertsDuringRead = value;
    }
    resetIgnoreVulnerabilityAlertsDuringRead() {
        this._ignoreVulnerabilityAlertsDuringRead = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get ignoreVulnerabilityAlertsDuringReadInput() {
        return this._ignoreVulnerabilityAlertsDuringRead;
    }
    get isTemplate() {
        return this.getBooleanAttribute('is_template');
    }
    set isTemplate(value) {
        this._isTemplate = value;
    }
    resetIsTemplate() {
        this._isTemplate = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get isTemplateInput() {
        return this._isTemplate;
    }
    get licenseTemplate() {
        return this.getStringAttribute('license_template');
    }
    set licenseTemplate(value) {
        this._licenseTemplate = value;
    }
    resetLicenseTemplate() {
        this._licenseTemplate = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get licenseTemplateInput() {
        return this._licenseTemplate;
    }
    get mergeCommitMessage() {
        return this.getStringAttribute('merge_commit_message');
    }
    set mergeCommitMessage(value) {
        this._mergeCommitMessage = value;
    }
    resetMergeCommitMessage() {
        this._mergeCommitMessage = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get mergeCommitMessageInput() {
        return this._mergeCommitMessage;
    }
    get mergeCommitTitle() {
        return this.getStringAttribute('merge_commit_title');
    }
    set mergeCommitTitle(value) {
        this._mergeCommitTitle = value;
    }
    resetMergeCommitTitle() {
        this._mergeCommitTitle = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get mergeCommitTitleInput() {
        return this._mergeCommitTitle;
    }
    get name() {
        return this.getStringAttribute('name');
    }
    set name(value) {
        this._name = value;
    }
    // Temporarily expose input value. Use with caution.
    get nameInput() {
        return this._name;
    }
    // node_id - computed: true, optional: false, required: false
    get nodeId() {
        return this.getStringAttribute('node_id');
    }
    // primary_language - computed: true, optional: false, required: false
    get primaryLanguage() {
        return this.getStringAttribute('primary_language');
    }
    get private() {
        return this.getBooleanAttribute('private');
    }
    set private(value) {
        this._private = value;
    }
    resetPrivate() {
        this._private = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get privateInput() {
        return this._private;
    }
    // repo_id - computed: true, optional: false, required: false
    get repoId() {
        return this.getNumberAttribute('repo_id');
    }
    get squashMergeCommitMessage() {
        return this.getStringAttribute('squash_merge_commit_message');
    }
    set squashMergeCommitMessage(value) {
        this._squashMergeCommitMessage = value;
    }
    resetSquashMergeCommitMessage() {
        this._squashMergeCommitMessage = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get squashMergeCommitMessageInput() {
        return this._squashMergeCommitMessage;
    }
    get squashMergeCommitTitle() {
        return this.getStringAttribute('squash_merge_commit_title');
    }
    set squashMergeCommitTitle(value) {
        this._squashMergeCommitTitle = value;
    }
    resetSquashMergeCommitTitle() {
        this._squashMergeCommitTitle = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get squashMergeCommitTitleInput() {
        return this._squashMergeCommitTitle;
    }
    // ssh_clone_url - computed: true, optional: false, required: false
    get sshCloneUrl() {
        return this.getStringAttribute('ssh_clone_url');
    }
    // svn_url - computed: true, optional: false, required: false
    get svnUrl() {
        return this.getStringAttribute('svn_url');
    }
    get topics() {
        return cdktf.Fn.tolist(this.getListAttribute('topics'));
    }
    set topics(value) {
        this._topics = value;
    }
    resetTopics() {
        this._topics = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get topicsInput() {
        return this._topics;
    }
    get visibility() {
        return this.getStringAttribute('visibility');
    }
    set visibility(value) {
        this._visibility = value;
    }
    resetVisibility() {
        this._visibility = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get visibilityInput() {
        return this._visibility;
    }
    get vulnerabilityAlerts() {
        return this.getBooleanAttribute('vulnerability_alerts');
    }
    set vulnerabilityAlerts(value) {
        this._vulnerabilityAlerts = value;
    }
    resetVulnerabilityAlerts() {
        this._vulnerabilityAlerts = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get vulnerabilityAlertsInput() {
        return this._vulnerabilityAlerts;
    }
    get pages() {
        return this._pages;
    }
    putPages(value) {
        this._pages.internalValue = value;
    }
    resetPages() {
        this._pages.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get pagesInput() {
        return this._pages.internalValue;
    }
    get securityAndAnalysis() {
        return this._securityAndAnalysis;
    }
    putSecurityAndAnalysis(value) {
        this._securityAndAnalysis.internalValue = value;
    }
    resetSecurityAndAnalysis() {
        this._securityAndAnalysis.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get securityAndAnalysisInput() {
        return this._securityAndAnalysis.internalValue;
    }
    get template() {
        return this._template;
    }
    putTemplate(value) {
        this._template.internalValue = value;
    }
    resetTemplate() {
        this._template.internalValue = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get templateInput() {
        return this._template.internalValue;
    }
    // =========
    // SYNTHESIS
    // =========
    synthesizeAttributes() {
        return {
            allow_auto_merge: cdktf.booleanToTerraform(this._allowAutoMerge),
            allow_merge_commit: cdktf.booleanToTerraform(this._allowMergeCommit),
            allow_rebase_merge: cdktf.booleanToTerraform(this._allowRebaseMerge),
            allow_squash_merge: cdktf.booleanToTerraform(this._allowSquashMerge),
            allow_update_branch: cdktf.booleanToTerraform(this._allowUpdateBranch),
            archive_on_destroy: cdktf.booleanToTerraform(this._archiveOnDestroy),
            archived: cdktf.booleanToTerraform(this._archived),
            auto_init: cdktf.booleanToTerraform(this._autoInit),
            default_branch: cdktf.stringToTerraform(this._defaultBranch),
            delete_branch_on_merge: cdktf.booleanToTerraform(this._deleteBranchOnMerge),
            description: cdktf.stringToTerraform(this._description),
            gitignore_template: cdktf.stringToTerraform(this._gitignoreTemplate),
            has_discussions: cdktf.booleanToTerraform(this._hasDiscussions),
            has_downloads: cdktf.booleanToTerraform(this._hasDownloads),
            has_issues: cdktf.booleanToTerraform(this._hasIssues),
            has_projects: cdktf.booleanToTerraform(this._hasProjects),
            has_wiki: cdktf.booleanToTerraform(this._hasWiki),
            homepage_url: cdktf.stringToTerraform(this._homepageUrl),
            id: cdktf.stringToTerraform(this._id),
            ignore_vulnerability_alerts_during_read: cdktf.booleanToTerraform(this._ignoreVulnerabilityAlertsDuringRead),
            is_template: cdktf.booleanToTerraform(this._isTemplate),
            license_template: cdktf.stringToTerraform(this._licenseTemplate),
            merge_commit_message: cdktf.stringToTerraform(this._mergeCommitMessage),
            merge_commit_title: cdktf.stringToTerraform(this._mergeCommitTitle),
            name: cdktf.stringToTerraform(this._name),
            private: cdktf.booleanToTerraform(this._private),
            squash_merge_commit_message: cdktf.stringToTerraform(this._squashMergeCommitMessage),
            squash_merge_commit_title: cdktf.stringToTerraform(this._squashMergeCommitTitle),
            topics: cdktf.listMapper(cdktf.stringToTerraform, false)(this._topics),
            visibility: cdktf.stringToTerraform(this._visibility),
            vulnerability_alerts: cdktf.booleanToTerraform(this._vulnerabilityAlerts),
            pages: repositoryPagesToTerraform(this._pages.internalValue),
            security_and_analysis: repositorySecurityAndAnalysisToTerraform(this._securityAndAnalysis.internalValue),
            template: repositoryTemplateToTerraform(this._template.internalValue),
        };
    }
}
exports.Repository = Repository;
_h = JSII_RTTI_SYMBOL_1;
Repository[_h] = { fqn: "@cdktf/provider-github.repository.Repository", version: "12.0.0" };
// =================
// STATIC PROPERTIES
// =================
Repository.tfResourceType = "github_repository";
