// @cdktf/provider-github/lib/repository-file/index.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.RepositoryFile = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const cdktf = require("cdktf");
/**
* Represents a {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/repository_file github_repository_file}
*/
class RepositoryFile extends cdktf.TerraformResource {
    // ==============
    // STATIC Methods
    // ==============
    /**
    * Generates CDKTF code for importing a RepositoryFile resource upon running "cdktf plan <stack-name>"
    * @param scope The scope in which to define this construct
    * @param importToId The construct id used in the generated config for the RepositoryFile to import
    * @param importFromId The id of the existing RepositoryFile that should be imported. Refer to the {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/repository_file#import import section} in the documentation of this resource for the id to use
    * @param provider? Optional instance of the provider where the RepositoryFile to import is found
    */
    static generateConfigForImport(scope, importToId, importFromId, provider) {
        return new cdktf.ImportableResource(scope, importToId, { terraformResourceType: "github_repository_file", importId: importFromId, provider });
    }
    // ===========
    // INITIALIZER
    // ===========
    /**
    * Create a new {@link https://registry.terraform.io/providers/integrations/github/5.40.0/docs/resources/repository_file github_repository_file} Resource
    *
    * @param scope The scope in which to define this construct
    * @param id The scoped construct ID. Must be unique amongst siblings in the same scope
    * @param options RepositoryFileConfig
    */
    constructor(scope, id, config) {
        super(scope, id, {
            terraformResourceType: 'github_repository_file',
            terraformGeneratorMetadata: {
                providerName: 'github',
                providerVersion: '5.40.0',
                providerVersionConstraint: '~> 5.0'
            },
            provider: config.provider,
            dependsOn: config.dependsOn,
            count: config.count,
            lifecycle: config.lifecycle,
            provisioners: config.provisioners,
            connection: config.connection,
            forEach: config.forEach
        });
        this._branch = config.branch;
        this._commitAuthor = config.commitAuthor;
        this._commitEmail = config.commitEmail;
        this._commitMessage = config.commitMessage;
        this._content = config.content;
        this._file = config.file;
        this._id = config.id;
        this._overwriteOnCreate = config.overwriteOnCreate;
        this._repository = config.repository;
    }
    get branch() {
        return this.getStringAttribute('branch');
    }
    set branch(value) {
        this._branch = value;
    }
    resetBranch() {
        this._branch = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get branchInput() {
        return this._branch;
    }
    get commitAuthor() {
        return this.getStringAttribute('commit_author');
    }
    set commitAuthor(value) {
        this._commitAuthor = value;
    }
    resetCommitAuthor() {
        this._commitAuthor = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get commitAuthorInput() {
        return this._commitAuthor;
    }
    get commitEmail() {
        return this.getStringAttribute('commit_email');
    }
    set commitEmail(value) {
        this._commitEmail = value;
    }
    resetCommitEmail() {
        this._commitEmail = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get commitEmailInput() {
        return this._commitEmail;
    }
    get commitMessage() {
        return this.getStringAttribute('commit_message');
    }
    set commitMessage(value) {
        this._commitMessage = value;
    }
    resetCommitMessage() {
        this._commitMessage = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get commitMessageInput() {
        return this._commitMessage;
    }
    // commit_sha - computed: true, optional: false, required: false
    get commitSha() {
        return this.getStringAttribute('commit_sha');
    }
    get content() {
        return this.getStringAttribute('content');
    }
    set content(value) {
        this._content = value;
    }
    // Temporarily expose input value. Use with caution.
    get contentInput() {
        return this._content;
    }
    get file() {
        return this.getStringAttribute('file');
    }
    set file(value) {
        this._file = value;
    }
    // Temporarily expose input value. Use with caution.
    get fileInput() {
        return this._file;
    }
    get id() {
        return this.getStringAttribute('id');
    }
    set id(value) {
        this._id = value;
    }
    resetId() {
        this._id = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get idInput() {
        return this._id;
    }
    get overwriteOnCreate() {
        return this.getBooleanAttribute('overwrite_on_create');
    }
    set overwriteOnCreate(value) {
        this._overwriteOnCreate = value;
    }
    resetOverwriteOnCreate() {
        this._overwriteOnCreate = undefined;
    }
    // Temporarily expose input value. Use with caution.
    get overwriteOnCreateInput() {
        return this._overwriteOnCreate;
    }
    // ref - computed: true, optional: false, required: false
    get ref() {
        return this.getStringAttribute('ref');
    }
    get repository() {
        return this.getStringAttribute('repository');
    }
    set repository(value) {
        this._repository = value;
    }
    // Temporarily expose input value. Use with caution.
    get repositoryInput() {
        return this._repository;
    }
    // sha - computed: true, optional: false, required: false
    get sha() {
        return this.getStringAttribute('sha');
    }
    // =========
    // SYNTHESIS
    // =========
    synthesizeAttributes() {
        return {
            branch: cdktf.stringToTerraform(this._branch),
            commit_author: cdktf.stringToTerraform(this._commitAuthor),
            commit_email: cdktf.stringToTerraform(this._commitEmail),
            commit_message: cdktf.stringToTerraform(this._commitMessage),
            content: cdktf.stringToTerraform(this._content),
            file: cdktf.stringToTerraform(this._file),
            id: cdktf.stringToTerraform(this._id),
            overwrite_on_create: cdktf.booleanToTerraform(this._overwriteOnCreate),
            repository: cdktf.stringToTerraform(this._repository),
        };
    }
}
exports.RepositoryFile = RepositoryFile;
_a = JSII_RTTI_SYMBOL_1;
RepositoryFile[_a] = { fqn: "@cdktf/provider-github.repositoryFile.RepositoryFile", version: "12.0.0" };
// =================
// STATIC PROPERTIES
// =================
RepositoryFile.tfResourceType = "github_repository_file";
